---
title: "R Notebook"
output: 
  html_document:
    toc: true
    toc_depth: 6
    toc_float: true
    code_folding: hide
    keep_md: true
    number_sections: true
---


``` r
library(here) 
```

```
## here() starts at C:/Users/fionn/Newcastle Grammar School/NGS SciX-2025 UoN DMG Cancer Signalling - Bioinformatics and Proteomics/Fionn Git/HSC-SciX-UoN_LS3.33
```

``` r
# Set global knitr chunk options
knitr::opts_chunk$set(
    error     = TRUE,        # Ensures that error messages are being shown in the output
    fig.align = "center",    # Ensures that all figures are aligned to the centre in the output
    message   = TRUE,       # <--- Recommend FALSE for cleaner reports, set to TRUE for development
    warning   = TRUE,       # <--- Recommend FALSE for cleaner reports, set to TRUE for development
    autodep   = TRUE,        # Automatically detects dependencies for caching
    cache     = TRUE,        # <--- IMPORTANT: Enable caching globally here!
    results   = "markup",    # Displays the output as normal formatted text
    echo      = TRUE,        # Ensures that the R code itself is displayed in the document
    dev       = "svg",       # Graphics output will be in Scalable Vector Graphics (SVG) format
    out.width = "90%",       # Sets the width of the output image or plot to 90% of the available width 
    out.extra = "keepaspectratio=true" # Adds extra options to the output, preserving aspect ratio
)
```


``` r
library(here)
```

# Set Up Working Environment

## Set Seed


``` r
seed <- "210825"
set.seed(seed)
message("seed has been set to 210825 ")
```

```
## seed has been set to 210825
```

## Load Packages & Functions


``` r
source(here("Custom R Functions and Scripts", "init_packages.R"))
```

```
## ── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
## ✔ dplyr     1.1.4     ✔ readr     2.1.5
## ✔ forcats   1.0.0     ✔ stringr   1.5.1
## ✔ ggplot2   3.5.2     ✔ tibble    3.3.0
## ✔ lubridate 1.9.4     ✔ tidyr     1.3.1
## ✔ purrr     1.1.0     
## ── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
## ✖ dplyr::filter() masks stats::filter()
## ✖ dplyr::lag()    masks stats::lag()
## ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
## 
## Attaching package: 'scales'
## 
## 
## The following object is masked from 'package:purrr':
## 
##     discard
## 
## 
## The following object is masked from 'package:readr':
## 
##     col_factor
## 
## 
## 
## Attaching package: 'kableExtra'
## 
## 
## The following object is masked from 'package:dplyr':
## 
##     group_rows
## 
## 
## Loading required package: SeuratObject
## 
## Loading required package: sp
## 
## 'SeuratObject' was built under R 4.5.0 but the current version is
## 4.5.1; it is recomended that you reinstall 'SeuratObject' as the ABI
## for R may have changed
## 
## 
## Attaching package: 'SeuratObject'
## 
## 
## The following object is masked from 'package:DT':
## 
##     JS
## 
## 
## The following objects are masked from 'package:base':
## 
##     intersect, t
## 
## 
## 
## Attaching package: 'Seurat'
## 
## 
## The following object is masked from 'package:DT':
## 
##     JS
## 
## 
## Loading required package: SummarizedExperiment
## 
## Loading required package: MatrixGenerics
## 
## Loading required package: matrixStats
## 
## 
## Attaching package: 'matrixStats'
## 
## 
## The following object is masked from 'package:dplyr':
## 
##     count
## 
## 
## 
## Attaching package: 'MatrixGenerics'
## 
## 
## The following objects are masked from 'package:matrixStats':
## 
##     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
##     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
##     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
##     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
##     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
##     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
##     colWeightedMeans, colWeightedMedians, colWeightedSds,
##     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
##     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
##     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
##     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
##     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
##     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
##     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
##     rowWeightedSds, rowWeightedVars
## 
## 
## Loading required package: GenomicRanges
## 
## Loading required package: stats4
## 
## Loading required package: BiocGenerics
## 
## Loading required package: generics
## 
## 
## Attaching package: 'generics'
## 
## 
## The following object is masked from 'package:lubridate':
## 
##     as.difftime
## 
## 
## The following object is masked from 'package:dplyr':
## 
##     explain
## 
## 
## The following objects are masked from 'package:base':
## 
##     as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,
##     setequal, union
## 
## 
## 
## Attaching package: 'BiocGenerics'
## 
## 
## The following object is masked from 'package:dplyr':
## 
##     combine
## 
## 
## The following objects are masked from 'package:stats':
## 
##     IQR, mad, sd, var, xtabs
## 
## 
## The following objects are masked from 'package:base':
## 
##     anyDuplicated, aperm, append, as.data.frame, basename, cbind,
##     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
##     get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,
##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
##     rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,
##     unsplit, which.max, which.min
## 
## 
## Loading required package: S4Vectors
## 
## 
## Attaching package: 'S4Vectors'
## 
## 
## The following objects are masked from 'package:lubridate':
## 
##     second, second<-
## 
## 
## The following objects are masked from 'package:dplyr':
## 
##     first, rename
## 
## 
## The following object is masked from 'package:tidyr':
## 
##     expand
## 
## 
## The following object is masked from 'package:utils':
## 
##     findMatches
## 
## 
## The following objects are masked from 'package:base':
## 
##     expand.grid, I, unname
## 
## 
## Loading required package: IRanges
## 
## 
## Attaching package: 'IRanges'
## 
## 
## The following object is masked from 'package:sp':
## 
##     %over%
## 
## 
## The following object is masked from 'package:glue':
## 
##     trim
## 
## 
## The following object is masked from 'package:lubridate':
## 
##     %within%
## 
## 
## The following objects are masked from 'package:dplyr':
## 
##     collapse, desc, slice
## 
## 
## The following object is masked from 'package:purrr':
## 
##     reduce
## 
## 
## The following object is masked from 'package:grDevices':
## 
##     windows
## 
## 
## Loading required package: GenomeInfoDb
## 
## Loading required package: Biobase
## 
## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     'browseVignettes()'. To cite Bioconductor, see
##     'citation("Biobase")', and for packages 'citation("pkgname")'.
## 
## 
## 
## Attaching package: 'Biobase'
## 
## 
## The following object is masked from 'package:MatrixGenerics':
## 
##     rowMedians
## 
## 
## The following objects are masked from 'package:matrixStats':
## 
##     anyMissing, rowMedians
## 
## 
## 
## Attaching package: 'SummarizedExperiment'
## 
## 
## The following object is masked from 'package:Seurat':
## 
##     Assays
## 
## 
## The following object is masked from 'package:SeuratObject':
## 
##     Assays
## 
## 
## Loading required package: SingleCellExperiment
## 
## Loading required package: scuttle
## 
## 
## Attaching package: 'monocle3'
## 
## 
## The following objects are masked from 'package:Biobase':
## 
##     exprs, fData, fData<-, pData, pData<-
## 
## 
## Loading required package: princurve
## 
## Loading required package: TrajectoryUtils
## 
## 
## Attaching package: 'TrajectoryUtils'
## 
## 
## The following object is masked from 'package:scran':
## 
##     createClusterMST
## 
## 
## Loading required package: Rcpp
## 
## Loading required package: data.table
## 
## 
## Attaching package: 'data.table'
## 
## 
## The following object is masked from 'package:SummarizedExperiment':
## 
##     shift
## 
## 
## The following object is masked from 'package:GenomicRanges':
## 
##     shift
## 
## 
## The following object is masked from 'package:IRanges':
## 
##     shift
## 
## 
## The following objects are masked from 'package:S4Vectors':
## 
##     first, second
## 
## 
## The following objects are masked from 'package:lubridate':
## 
##     hour, isoweek, mday, minute, month, quarter, second, wday, week,
##     yday, year
## 
## 
## The following objects are masked from 'package:dplyr':
## 
##     between, first, last
## 
## 
## The following object is masked from 'package:purrr':
## 
##     transpose
## 
## 
## 
## Attaching package: 'presto'
## 
## 
## The following object is masked from 'package:monocle3':
## 
##     top_markers
## 
## 
## 
## Attaching package: 'glmGamPoi'
## 
## 
## The following object is masked from 'package:dplyr':
## 
##     vars
## 
## 
## The following object is masked from 'package:ggplot2':
## 
##     vars
## 
## 
## 
## Attaching package: 'edge'
## 
## 
## The following object is masked from 'package:monocle3':
## 
##     fit_models
## 
## 
## 
## Attaching package: 'scrapper'
## 
## 
## The following object is masked from 'package:scran':
## 
##     scoreMarkers
## 
## 
## The following objects are masked from 'package:scater':
## 
##     aggregateAcrossCells, normalizeCounts
## 
## 
## The following objects are masked from 'package:scuttle':
## 
##     aggregateAcrossCells, normalizeCounts
## 
## 
## 
## Attaching package: 'DESeq2'
## 
## 
## The following object is masked from 'package:scater':
## 
##     fpkm
## 
## 
## Loading required package: limma
## 
## 
## Attaching package: 'limma'
## 
## 
## The following object is masked from 'package:DESeq2':
## 
##     plotMA
## 
## 
## The following object is masked from 'package:scater':
## 
##     plotMDS
## 
## 
## The following object is masked from 'package:BiocGenerics':
## 
##     plotMA
## 
## 
## 
## Attaching package: 'edgeR'
## 
## 
## The following object is masked from 'package:SingleCellExperiment':
## 
##     cpm
## 
## 
## 
## 
## clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/
## 
## Please cite:
## 
## Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.
## clusterProfiler: an R package for comparing biological themes among
## gene clusters. OMICS: A Journal of Integrative Biology. 2012,
## 16(5):284-287
## 
## 
## Attaching package: 'clusterProfiler'
## 
## 
## The following object is masked from 'package:IRanges':
## 
##     slice
## 
## 
## The following object is masked from 'package:S4Vectors':
## 
##     rename
## 
## 
## The following object is masked from 'package:purrr':
## 
##     simplify
## 
## 
## The following object is masked from 'package:stats':
## 
##     filter
## 
## 
## Loading required package: AnnotationDbi
## 
## 
## Attaching package: 'AnnotationDbi'
## 
## 
## The following object is masked from 'package:clusterProfiler':
## 
##     select
## 
## 
## The following object is masked from 'package:dplyr':
## 
##     select
## 
## 
## 
## 
## 
## Attaching package: 'celldex'
## 
## 
## The following objects are masked from 'package:SingleR':
## 
##     BlueprintEncodeData, DatabaseImmuneCellExpressionData,
##     HumanPrimaryCellAtlasData, ImmGenData, MonacoImmuneData,
##     MouseRNAseqData, NovershternHematopoieticData
## 
## 
## 
## Attaching package: 'cowplot'
## 
## 
## The following object is masked from 'package:patchwork':
## 
##     align_plots
## 
## 
## The following object is masked from 'package:ggpubr':
## 
##     get_legend
## 
## 
## The following object is masked from 'package:lubridate':
## 
##     stamp
## 
## 
## 
## Attaching package: 'gridExtra'
## 
## 
## The following object is masked from 'package:Biobase':
## 
##     combine
## 
## 
## The following object is masked from 'package:BiocGenerics':
## 
##     combine
## 
## 
## The following object is masked from 'package:dplyr':
## 
##     combine
## 
## 
## Loading required package: viridisLite
## 
## 
## Attaching package: 'viridis'
## 
## 
## The following object is masked from 'package:scales':
## 
##     viridis_pal
## 
## 
## Loading required package: grid
## 
## ========================================
## ComplexHeatmap version 2.25.2
## Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/
## Github page: https://github.com/jokergoo/ComplexHeatmap
## Documentation: http://jokergoo.github.io/ComplexHeatmap-reference
## 
## If you use it in published research, please cite either one:
## - Gu, Z. Complex Heatmap Visualization. iMeta 2022.
## - Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional 
##     genomic data. Bioinformatics 2016.
## 
## 
## The new InteractiveComplexHeatmap package can directly export static 
## complex heatmaps into an interactive Shiny app with zero effort. Have a try!
## 
## This message can be suppressed by:
##   suppressPackageStartupMessages(library(ComplexHeatmap))
## ========================================
## 
## 
## 
## Attaching package: 'pheatmap'
## 
## 
## The following object is masked from 'package:ComplexHeatmap':
## 
##     pheatmap
## 
## 
## ========================================
## circlize version 0.4.16
## CRAN page: https://cran.r-project.org/package=circlize
## Github page: https://github.com/jokergoo/circlize
## Documentation: https://jokergoo.github.io/circlize_book/book/
## 
## If you use it in published research, please cite:
## Gu, Z. circlize implements and enhances circular visualization
##   in R. Bioinformatics 2014.
## 
## This message can be suppressed by:
##   suppressPackageStartupMessages(library(circlize))
## ========================================
## 
## 
## Loading required package: futile.logger
## 
## 
## Attaching package: 'VennDiagram'
## 
## 
## The following object is masked from 'package:ggpubr':
## 
##     rotate
## 
## 
## 
## Attaching package: 'hdf5r'
## 
## 
## The following object is masked from 'package:SummarizedExperiment':
## 
##     values
## 
## 
## The following object is masked from 'package:GenomicRanges':
## 
##     values
## 
## 
## The following object is masked from 'package:S4Vectors':
## 
##     values
## 
## 
## The following object is masked from 'package:purrr':
## 
##     flatten_df
## 
## 
## Loading required package: igraph
## 
## 
## Attaching package: 'igraph'
## 
## 
## The following object is masked from 'package:circlize':
## 
##     degree
## 
## 
## The following object is masked from 'package:clusterProfiler':
## 
##     simplify
## 
## 
## The following object is masked from 'package:monocle3':
## 
##     clusters
## 
## 
## The following object is masked from 'package:scater':
## 
##     normalize
## 
## 
## The following object is masked from 'package:GenomicRanges':
## 
##     union
## 
## 
## The following object is masked from 'package:IRanges':
## 
##     union
## 
## 
## The following object is masked from 'package:S4Vectors':
## 
##     union
## 
## 
## The following objects are masked from 'package:BiocGenerics':
## 
##     normalize, path, union
## 
## 
## The following objects are masked from 'package:generics':
## 
##     components, union
## 
## 
## The following object is masked from 'package:Seurat':
## 
##     components
## 
## 
## The following objects are masked from 'package:lubridate':
## 
##     %--%, union
## 
## 
## The following objects are masked from 'package:dplyr':
## 
##     as_data_frame, groups, union
## 
## 
## The following objects are masked from 'package:purrr':
## 
##     compose, simplify
## 
## 
## The following object is masked from 'package:tidyr':
## 
##     crossing
## 
## 
## The following object is masked from 'package:tibble':
## 
##     as_data_frame
## 
## 
## The following objects are masked from 'package:stats':
## 
##     decompose, spectrum
## 
## 
## The following object is masked from 'package:base':
## 
##     union
## 
## 
## Registered S3 methods overwritten by 'registry':
##   method               from 
##   print.registry_field proxy
##   print.registry_entry proxy
## 
## Registered S3 method overwritten by 'ggnetwork':
##   method         from  
##   fortify.igraph ggtree
## 
## 
## Attaching package: 'CellChat'
## 
## 
## The following objects are masked from 'package:scater':
## 
##     runPCA, runUMAP
## 
## 
## 
## Attaching package: 'future'
## 
## 
## The following objects are masked from 'package:igraph':
## 
##     %->%, %<-%
```

## Parallelisation


``` r
# Set up parallel processing for CellChat or other parallelized tasks
library(future)
plan("multisession", workers = 10)
options(future.globals.maxSize = 10240 * 1024^2)  # 10 GB memory cap
```

## Configuration

### Custom Colours


``` r
# Custom colors
man_cols <- c("#86b0cc",  "#f3e65d", "#d5c1e7", "#eeb84c", "#82c39e", "#525252","#4d9f6b", "#b3939e", "#e76031", "#e9944b")
names(man_cols) <- c("B_cells", "NK_cells", "monocytes", "T_cells", "neutrophils", "megakaryocytes", "pDCs",
                     "plasma_cells", "progenitor_cells", "NKT_cells")
```

## Output Folders

### Primary


``` r
dir.create(here("R Projects", "2907", "output"), recursive = TRUE, showWarnings = FALSE)
```

### DGE


``` r
dir.create(here("R Projects", "2907", "output", "de_analysis", "Find Markers"), recursive = TRUE, showWarnings = FALSE)
dir.create(here("R Projects", "2907", "output", "de_analysis", "Presto"), recursive = TRUE, showWarnings = FALSE)
dir.create(here("R Projects", "2907", "output", "de_analysis", "t.test"), recursive = TRUE, showWarnings = FALSE)
```

### Visualisation


``` r
dir.create(here("R Projects", "2907", "output", "visualisation", "box plots"), recursive = TRUE, showWarnings = FALSE)
dir.create(here("R Projects", "2907", "output", "visualisation", "complex heatmaps"), recursive = TRUE, showWarnings = FALSE)
dir.create(here("R Projects", "2907", "output", "visualisation", "nebulosa"), recursive = TRUE, showWarnings = FALSE)
dir.create(here("R Projects", "2907", "output", "visualisation", "violin plots"), recursive = TRUE, showWarnings = FALSE)
dir.create(here("R Projects", "2907", "output", "visualisation", "volcano"), recursive = TRUE, showWarnings = FALSE)
```

## Import & Pre-processing of Data

### Load RDS file

#### Load Original RDS

Importing PhD Clara Savary's RDS & Initialising the object as a Seurat Object


``` r
seu <- readRDS(here("Data",          "seu_NKT_blood_mouse_PPK_ONC_harmony_integrated_filtered_v2_20231_20236_annot.rds"))
```

#### Subset

Initial Subset to isolate T Cells, Natural Killer T Cells and Natural Killer Cells and label


``` r
seu_NKT <- subset(seu, subset = cell_types %in% c("T_cells", "NK_cells", "NKT_cells"))
```

#### Subset Dataset Again

Second Subset -\> subsetting to remove treatment conditions involving theraputics as this is outside the bounds of the scope of the investigation.

Including:

-   DMG Mice (represented as untreated or UT)

-   Control: No Cancer (Treatment and Disease Naive Group)

-   Surgery Control (SHAM - Injected with Saline - to make sure that the stress of surgery isn't a factor and can be compared between Naive, UT and Sham)

Excluding:

-   ONC201 - Dordaviprone -\> DRD2 antagonist and ClpP agonist

    -   <https://doi.org/10.1158/0008-5472.CAN-23-0186>

    -   <https://doi.org/10.1158/2159-8290.CD-23-0131>

-   Dex - Dexmethasone -\> Corticosteroid

-   Dex & ONC201 Combo


``` r
seu_NKT_focused <- subset(seu_NKT, subset = treatment %in% c("UT", "NAIVE", "SHAM"))
```

For later scripting - load seu_NKT_focused

#### Export

Use either base or BioCgenerics

##### BioCgenerics


``` r
BiocGenerics::saveRDS(seu_NKT_focused, file = here("Data", "seu_NKT_focused.rds"))
```

OR

##### Base


``` r
base::saveRDS(seu_NKT_focused, file = here("Data", "seu_NKT_focused.rds") )
```

#### Load rds


``` r
seu_NKT_focused <- readRDS(here("Data", "seu_NKT_focused.rds"))
```

### T Cell Subset


``` r
seu_T <- subset(seu_NKT_focused, subset = T_clusters %in% "T_cells")
```

# QC

Already completed by PhD CS

# Cell Type Annotation


``` r
# UMAP plot showing cluster identities
p_ident <- DimPlot(
  seu_NKT_focused,
  label     = TRUE,
  repel     = TRUE,
  group.by  = "T_clusters",
  reduction = "umap_harmony_SCT"
) + NoLegend()

# Display the plot
print(p_ident)
```

<img src="2907_files/figure-html/unnamed-chunk-17-1.svg" width="90%" keepaspectratio=true style="display: block; margin: auto;" />


``` r
# Extract RNA assay data to use as SingleR test input
expression <- GetAssayData(seu_NKT_focused[["RNA"]], layer = "data")

# Use tree-based cluster identities for annotation
cluster_info <- seu_NKT_focused@meta.data$T_clusters

# Load reference atlases for annotation
atlas_set <- list(
  "MRD"    = celldex::MouseRNAseqData(),
  "DICED"  = celldex::DatabaseImmuneCellExpressionData(),
  "IMMGEN" = celldex::ImmGenData()
)

# Loop over each reference set and label level
sapply(names(atlas_set), function(x) {
  
  for (label in c("label.main", "label.fine")) {
    
    # Run SingleR on clustered data
    prediction <- SingleR(
      test     = expression,
      ref      = atlas_set[[x]],
      labels   = atlas_set[[x]][[label]],
      clusters = cluster_info
    )

    # Plot annotation scores as a heatmap
    p_singleR <- plotScoreHeatmap(prediction, show_colnames = TRUE) %>% as.ggplot()

    # Display UMAP and annotation heatmap side-by-side
    p <- grid.arrange(grobs = list(p_ident, p_singleR), ncol = 2)
    p
    
  }
  
})
```

<img src="2907_files/figure-html/unnamed-chunk-18-1.svg" width="90%" keepaspectratio=true style="display: block; margin: auto;" /><img src="2907_files/figure-html/unnamed-chunk-18-2.svg" width="90%" keepaspectratio=true style="display: block; margin: auto;" /><img src="2907_files/figure-html/unnamed-chunk-18-3.svg" width="90%" keepaspectratio=true style="display: block; margin: auto;" /><img src="2907_files/figure-html/unnamed-chunk-18-4.svg" width="90%" keepaspectratio=true style="display: block; margin: auto;" /><img src="2907_files/figure-html/unnamed-chunk-18-5.svg" width="90%" keepaspectratio=true style="display: block; margin: auto;" /><img src="2907_files/figure-html/unnamed-chunk-18-6.svg" width="90%" keepaspectratio=true style="display: block; margin: auto;" /><img src="2907_files/figure-html/unnamed-chunk-18-7.svg" width="90%" keepaspectratio=true style="display: block; margin: auto;" /><img src="2907_files/figure-html/unnamed-chunk-18-8.svg" width="90%" keepaspectratio=true style="display: block; margin: auto;" /><img src="2907_files/figure-html/unnamed-chunk-18-9.svg" width="90%" keepaspectratio=true style="display: block; margin: auto;" /><img src="2907_files/figure-html/unnamed-chunk-18-10.svg" width="90%" keepaspectratio=true style="display: block; margin: auto;" /><img src="2907_files/figure-html/unnamed-chunk-18-11.svg" width="90%" keepaspectratio=true style="display: block; margin: auto;" /><img src="2907_files/figure-html/unnamed-chunk-18-12.svg" width="90%" keepaspectratio=true style="display: block; margin: auto;" />

```
## $MRD
## NULL
## 
## $DICED
## NULL
## 
## $IMMGEN
## NULL
```

# Dim Reduction & Clustering

Clara already created Dim Plots defined in her Seurat Object - all we have to do now is run them if we need to make more (we don't) we can


``` r
# RNA Assay ====
## PCA ====
DimPlot(seu_NKT_focused, reduction = "pca_RNA")
```

<img src="2907_files/figure-html/unnamed-chunk-19-1.svg" width="90%" keepaspectratio=true style="display: block; margin: auto;" />

``` r
## UMAP ====
DimPlot(seu_NKT_focused, reduction = "umap_RNA")
```

<img src="2907_files/figure-html/unnamed-chunk-19-2.svg" width="90%" keepaspectratio=true style="display: block; margin: auto;" />

``` r
## TSNE ====
DimPlot(seu_NKT_focused, reduction = "tsne_RNA")
```

<img src="2907_files/figure-html/unnamed-chunk-19-3.svg" width="90%" keepaspectratio=true style="display: block; margin: auto;" />

``` r
# PCA
DimPlot(seu_NKT_focused, reduction = "pca")
```

<img src="2907_files/figure-html/unnamed-chunk-19-4.svg" width="90%" keepaspectratio=true style="display: block; margin: auto;" />

``` r
# SCT - SC Transform (Normalisation Technique) Assay ====
## UMAP SCT ====
DimPlot(seu_NKT_focused, reduction = "umap_SCT")
```

<img src="2907_files/figure-html/unnamed-chunk-19-5.svg" width="90%" keepaspectratio=true style="display: block; margin: auto;" />

``` r
## Harmony SCT ====
DimPlot(seu_NKT_focused, reduction = "harmony_SCT")
```

<img src="2907_files/figure-html/unnamed-chunk-19-6.svg" width="90%" keepaspectratio=true style="display: block; margin: auto;" />

``` r
## UMAP Harmony SCT ====
DimPlot(seu_NKT_focused, reduction = "umap_harmony_SCT")
```

<img src="2907_files/figure-html/unnamed-chunk-19-7.svg" width="90%" keepaspectratio=true style="display: block; margin: auto;" />

# Differential Expression Analysis

MAST, Find Markers, Presto and deseq2/edgeR

# T Clusters

Using the Seu_T Seurat Object to exclude NKT, NK Cells

## Presto

### Wilcoxon


``` r
#Define the gene list 
selected_genes <- c("S1pr1", "S1pr2", "S1pr3", "S1pr4", "S1pr5",
                    "Ccr7", "Sell", "Cd69", "Klf2", "Cxcr4", "Cd44", "Itgae", "Itgal")

pairwise_comparisons <- list(
  "NAIVE_WB_vs_NAIVE_BM"     = c("Naive_WB", "Naive_BM"),
  "NAIVE_WB_vs_SHAM_WB"      = c("Naive_WB", "Sham_WB"),
  "NAIVE_WB_vs_SHAM_BM"      = c("Naive_WB", "Sham_BM"),
  "NAIVE_WB_vs_UT_WB"        = c("Naive_WB", "UT_WB"),
  "NAIVE_WB_vs_UT_BM"        = c("Naive_WB", "UT_BM"),
  "NAIVE_BM_vs_SHAM_WB"      = c("Naive_BM", "Sham_WB"),
  "NAIVE_BM_vs_SHAM_BM"      = c("Naive_BM", "Sham_BM"),
  "NAIVE_BM_vs_UT_WB"        = c("Naive_BM", "UT_WB"),
  "NAIVE_BM_vs_UT_BM"        = c("Naive_BM", "UT_BM"),
  "SHAM_WB_vs_SHAM_BM"       = c("Sham_WB", "Sham_BM"),
  "SHAM_WB_vs_UT_WB"         = c("Sham_WB", "UT_WB"),
  "SHAM_WB_vs_UT_BM"         = c("Sham_WB", "UT_BM"),
  "SHAM_BM_vs_UT_WB"         = c("Sham_BM", "UT_WB"),
  "SHAM_BM_vs_UT_BM"         = c("Sham_BM", "UT_BM"),
  "UT_WB_vs_UT_BM"           = c("UT_WB", "UT_BM")
)

wb <- createWorkbook()

for (i in names(pairwise_comparisons)) {
  
  deg_T <- wilcoxauc(seu_T, group_by = "sample_name", groups_use = pairwise_comparisons[[i]])
  
  addWorksheet(wb, sheetName = i)
  
  deg_T <- deg_T %>%
    filter(group == pairwise_comparisons[[i]][[1]]) %>%
    filter(feature %in% selected_genes) %>%   # <- Filter only your selected genes
    arrange(group, -logFC)
  
  writeData(wb, sheet = i, x = deg_T)
}
```

```
## Warning: The `slot` argument of `GetAssayData()` is deprecated as of SeuratObject 5.0.0.
## ℹ Please use the `layer` argument instead.
## ℹ The deprecated feature was likely used in the presto package.
##   Please report the issue to the authors.
## This warning is displayed once every 8 hours.
## Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
## generated.
```

``` r
saveWorkbook(wb, here("R Projects", "2907", "output", "de_analysis", "Presto", "DE_presto-wilcoxauc_filtered.xlsx"), overwrite = TRUE)
```

## Find Markers

#### Wilcoxon Limma


``` r
# Define the gene list you want to focus on
selected_genes <- c("S1pr1", "S1pr2", "S1pr3", "S1pr4", "S1pr5", 
                    "Ccr7", "Sell", "Cd69", "Klf2", "Cxcr4", "Cd44", "Itgae", "Itgal")

pairwise_comparisons <- list(
  "NAIVE_WB_vs_NAIVE_BM"     = c("Naive_WB", "Naive_BM"),
  "NAIVE_WB_vs_SHAM_WB"      = c("Naive_WB", "Sham_WB"),
  "NAIVE_WB_vs_SHAM_BM"      = c("Naive_WB", "Sham_BM"),
  "NAIVE_WB_vs_UT_WB"        = c("Naive_WB", "UT_WB"),
  "NAIVE_WB_vs_UT_BM"        = c("Naive_WB", "UT_BM"),
  "NAIVE_BM_vs_SHAM_WB"      = c("Naive_BM", "Sham_WB"),
  "NAIVE_BM_vs_SHAM_BM"      = c("Naive_BM", "Sham_BM"),
  "NAIVE_BM_vs_UT_WB"        = c("Naive_BM", "UT_WB"),
  "NAIVE_BM_vs_UT_BM"        = c("Naive_BM", "UT_BM"),
  "SHAM_WB_vs_SHAM_BM"       = c("Sham_WB", "Sham_BM"),
  "SHAM_WB_vs_UT_WB"         = c("Sham_WB", "UT_WB"),
  "SHAM_WB_vs_UT_BM"         = c("Sham_WB", "UT_BM"),
  "SHAM_BM_vs_UT_WB"         = c("Sham_BM", "UT_WB"),
  "SHAM_BM_vs_UT_BM"         = c("Sham_BM", "UT_BM"),
  "UT_WB_vs_UT_BM"           = c("UT_WB", "UT_BM")
)

wb <- createWorkbook()

for (i in names(pairwise_comparisons)) {
  
  deg_T <- FindMarkers(seu_T, group.by = "sample_name", ident.1 = pairwise_comparisons[[i]][1], ident.2 = pairwise_comparisons[[i]][2], test.use = "wilcox_limma")
  
  addWorksheet(wb, sheetName = i)
  
  deg_T <- deg_T %>%
    filter(rownames(.) %in% selected_genes) %>%   # <- Filter only your selected genes
    arrange(-avg_log2FC)
  
  # Add gene names as a column
  deg_T$gene <- rownames(deg_T)
  deg_T <- deg_T %>% dplyr::select(gene, dplyr::everything())
  
  writeData(wb, sheet = i, x = deg_T)
}
```

```
## Warning: `PackageCheck()` was deprecated in SeuratObject 5.0.0.
## ℹ Please use `rlang::check_installed()` instead.
## ℹ The deprecated feature was likely used in the Seurat package.
##   Please report the issue at <https://github.com/satijalab/seurat/issues>.
## This warning is displayed once every 8 hours.
## Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
## generated.
```

``` r
saveWorkbook(wb, here("R Projects", "2907", "output", "de_analysis", "Find Markers", "DE_wilcox-limma_filtered.xlsx"), overwrite = TRUE)
```

#### deseq2


``` r
# Create output directories
dir.create(here("R Projects", "2907", "output", "visualisation", "enhanced_volcano"), recursive = TRUE, showWarnings = FALSE)
dir.create(here("R Projects", "2907", "output", "visualisation", "enhanced_volcano", "deseq2"), recursive = TRUE, showWarnings = FALSE)
dir.create(here("R Projects", "2907", "output", "visualisation", "enhanced_volcano", "edgeR"), recursive = TRUE, showWarnings = FALSE)

# Define the gene list you want to focus on
selected_genes <- c("S1pr1", "S1pr2", "S1pr3", "S1pr4", "S1pr5", 
                    "Ccr7", "Sell", "Cd69", "Klf2", "Cxcr4", "Cd44", "Itgae", "Itgal")

pairwise_comparisons <- list(
  "NAIVE_WB_vs_NAIVE_BM"     = c("Naive_WB", "Naive_BM"),
  "NAIVE_WB_vs_SHAM_WB"      = c("Naive_WB", "Sham_WB"),
  "NAIVE_WB_vs_SHAM_BM"      = c("Naive_WB", "Sham_BM"),
  "NAIVE_WB_vs_UT_WB"        = c("Naive_WB", "UT_WB"),
  "NAIVE_WB_vs_UT_BM"        = c("Naive_WB", "UT_BM"),
  "NAIVE_BM_vs_SHAM_WB"      = c("Naive_BM", "Sham_WB"),
  "NAIVE_BM_vs_SHAM_BM"      = c("Naive_BM", "Sham_BM"),
  "NAIVE_BM_vs_UT_WB"        = c("Naive_BM", "UT_WB"),
  "NAIVE_BM_vs_UT_BM"        = c("Naive_BM", "UT_BM"),
  "SHAM_WB_vs_SHAM_BM"       = c("Sham_WB", "Sham_BM"),
  "SHAM_WB_vs_UT_WB"         = c("Sham_WB", "UT_WB"),
  "SHAM_WB_vs_UT_BM"         = c("Sham_WB", "UT_BM"),
  "SHAM_BM_vs_UT_WB"         = c("Sham_BM", "UT_WB"),
  "SHAM_BM_vs_UT_BM"         = c("Sham_BM", "UT_BM"),
  "UT_WB_vs_UT_BM"           = c("UT_WB", "UT_BM")
)

# Function to run DESeq2 analysis for a pairwise comparison
run_DESeq2_comparison <- function(counts_matrix, sample_info, group1, group2, comparison_name) {
  
  cat("\nRunning DESeq2 analysis for:", comparison_name, "\n")
  
  # Filter samples for this comparison
  samples_to_use <- sample_info$sample[sample_info$group %in% c(group1, group2)]
  
  if (length(samples_to_use) < 4) {  # Need at least 2 replicates per group
    cat("Warning: Not enough samples for comparison", comparison_name, "- found", length(samples_to_use), "samples\n")
    return(NULL)
  }
  
  # Subset counts and sample info
  counts_subset <- counts_matrix[, samples_to_use, drop = FALSE]
  sample_info_subset <- sample_info[sample_info$sample %in% samples_to_use, ]
  
  # Ensure counts are integers for DESeq2
  counts_subset <- round(counts_subset)
  
  # Create colData for DESeq2
  colData <- data.frame(
    sample = sample_info_subset$sample,
    group = factor(sample_info_subset$group, levels = c(group2, group1)), # Reference level first
    row.names = sample_info_subset$sample
  )
  
  cat("Groups:", levels(colData$group), "\n")
  cat("Reference group:", group2, "\n")
  
  # Create DESeq2 dataset
  dds <- DESeqDataSetFromMatrix(
    countData = counts_subset,
    colData = colData,
    design = ~ group
  )
  
  # Filter low count genes
  keep <- rowSums(counts(dds)) >= 10
  dds <- dds[keep,]
  
  cat("Kept", sum(keep), "genes after filtering\n")
  
  # Run DESeq2 analysis
  dds <- DESeq(dds, quiet = TRUE)
  
  # Get results (group1 vs group2, with group2 as reference)
  res <- results(dds, contrast = c("group", group1, group2), alpha = 0.05)
  
  # Convert to data frame
  results_df <- as.data.frame(res)
  results_df$gene <- rownames(results_df)
  
  # Filter for selected genes only
  results_filtered <- results_df[results_df$gene %in% selected_genes, ]
  
  # Rename columns for consistency with other methods
  results_filtered <- results_filtered %>%
    dplyr::select(gene, log2FoldChange, baseMean, pvalue, padj) %>%
    rename(avg_log2FC = log2FoldChange, p_val = pvalue, p_val_adj = padj) %>%
    arrange(-avg_log2FC) %>%
    filter(!is.na(avg_log2FC))  # Remove genes with NA fold changes
  
  cat("Found", nrow(results_filtered), "genes from selected list\n")
  
  return(results_filtered)
}

# Create pseudobulk counts (reuse from edgeR analysis if available, otherwise create new)
if (!exists("pseudobulk_counts") || !exists("sample_info")) {
  cat("Creating pseudobulk counts matrix with replicates for DESeq2...\n")
  source(here("Custom R Functions and Scripts", "create_pseudobulk.R"))  # Assuming you saved the function
  pseudobulk_counts <- create_pseudobulk_with_replicates(seu_T, "sample_name", n_replicates = 3)
  
  # Create sample info
  sample_info <- data.frame(
    sample = colnames(pseudobulk_counts),
    stringsAsFactors = FALSE
  )
  sample_info$original_sample <- sapply(strsplit(sample_info$sample, "_rep"), function(x) x[1])
  sample_info$treatment <- sapply(strsplit(sample_info$original_sample, "_"), function(x) x[1])
  sample_info$tissue <- sapply(strsplit(sample_info$original_sample, "_"), function(x) x[2])
  sample_info$group <- sample_info$original_sample
} else {
  cat("Using existing pseudobulk counts matrix...\n")
}
```

```
## Creating pseudobulk counts matrix with replicates for DESeq2...
```

```
## Warning in file(filename, "r", encoding = encoding): cannot open file
## 'C:/Users/fionn/Newcastle Grammar School/NGS SciX-2025 UoN DMG Cancer
## Signalling - Bioinformatics and Proteomics/Fionn Git/HSC-SciX-UoN_LS3.33/Custom
## R Functions and Scripts/create_pseudobulk.R': No such file or directory
```

```
## Error in file(filename, "r", encoding = encoding): cannot open the connection
```

``` r
# Initialize Excel workbook for DESeq2 results
wb_deseq2 <- createWorkbook()

# Run DESeq2 analysis and generate volcano plots for each comparison
for (i in names(pairwise_comparisons)) {
  
  group1 <- pairwise_comparisons[[i]][1]
  group2 <- pairwise_comparisons[[i]][2]
  
  cat("\n", rep("=", 50), "\n")
  cat("Processing DESeq2:", i, "\n")
  cat(rep("=", 50), "\n")
  
  # Run DESeq2 analysis for this comparison
  deg_T <- run_DESeq2_comparison(pseudobulk_counts, sample_info, group1, group2, i)
  
  if (!is.null(deg_T) && nrow(deg_T) > 0) {
    
    # Add to Excel workbook
    addWorksheet(wb_deseq2, sheetName = i)
    writeData(wb_deseq2, sheet = i, x = deg_T)
    
    # Create volcano plot
    p <- EnhancedVolcano(
      deg_T,
      lab = deg_T$gene,
      x = "avg_log2FC",
      y = "p_val_adj",
      selectLab = intersect(deg_T$gene, selected_genes),
      pCutoff = 0.05,
      FCcutoff = 0.5,
      title = paste("DESeq2:", i),
      subtitle = paste("Pseudobulk analysis -", group1, "vs", group2),
      caption = paste("Total genes in plot:", nrow(deg_T)),
      pointSize = 3.0,
      labSize = 4.5,
      drawConnectors = TRUE,
      widthConnectors = 0.5,
      colConnectors = "grey50",
      legendPosition = "right",
      legendLabSize = 12,
      legendIconSize = 4.0,
      col = c("grey30", "forestgreen", "royalblue", "red2"),
      colAlpha = 0.7,
      xlim = c(min(deg_T$avg_log2FC, na.rm = TRUE) - 0.5, 
               max(deg_T$avg_log2FC, na.rm = TRUE) + 0.5),
      ylim = c(0, max(-log10(deg_T$p_val_adj), na.rm = TRUE) + 1),
      gridlines.major = TRUE,
      gridlines.minor = FALSE,
      border = "full",
      borderWidth = 1.0,
      borderColour = "black"
    )
    
    # Add custom theme
    p <- p + 
      theme_bw() +
      theme(
        plot.title = element_text(size = 14, face = "bold", hjust = 0.5),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        axis.title = element_text(size = 12, face = "bold"),
        axis.text = element_text(size = 11),
        legend.title = element_text(size = 12, face = "bold"),
        legend.text = element_text(size = 11),
        panel.grid.major = element_line(color = "grey90", size = 0.5),
        panel.border = element_rect(color = "black", fill = NA, size = 1)
      )
    
    # Display the plot
    print(p)
    
    # Save to DESeq2 subfolder
    ggsave(
      filename = file.path(here("R Projects", "2907", "output", "visualisation", "enhanced_volcano", "deseq2"), 
                          paste0(i, "_DESeq2_pseudobulk.svg")),
      plot = p, width = 14, height = 10, dpi = 300, bg = "white"
    )
    
    ggsave(
      filename = file.path(here("R Projects", "2907", "output", "visualisation", "enhanced_volcano", "deseq2"), 
                          paste0(i, "_DESeq2_pseudobulk.png")),
      plot = p, width = 14, height = 10, dpi = 300, bg = "white"
    )
    
    cat("✓ DESeq2 volcano plot created and saved for:", i, "\n")
    cat("  - Genes in plot:", nrow(deg_T), "\n")
    cat("  - Significant genes (FDR < 0.05):", sum(deg_T$p_val_adj < 0.05, na.rm = TRUE), "\n")
    
  } else {
    cat("✗ No DESeq2 results generated for:", i, "\n")
  }
}
```

```
## 
##  = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = 
## Processing DESeq2: NAIVE_WB_vs_NAIVE_BM 
## = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = 
## 
## Running DESeq2 analysis for: NAIVE_WB_vs_NAIVE_BM
```

```
## Error: object 'sample_info' not found
```

``` r
# Save DESeq2 Excel results
saveWorkbook(wb_deseq2, here("R Projects", "2907", "output", "de_analysis", "Find Markers", "DE_DESeq2_pseudobulk_filtered.xlsx"), overwrite = TRUE)
```

```
## Warning: Workbook does not contain any worksheets. A worksheet will be added.
```

``` r
cat("\n" , rep("=", 60), "\n")
```

```
## 
##  = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = =
```

``` r
cat("DESEQ2 VOLCANO PLOT GENERATION COMPLETED!\n")
```

```
## DESEQ2 VOLCANO PLOT GENERATION COMPLETED!
```

``` r
cat(rep("=", 60), "\n")
```

```
## = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = =
```

``` r
cat("All DESeq2 plots saved to:", here("R Projects", "2907", "output", "visualisation", "enhanced_volcano", "deseq2"), "\n")
```

```
## All DESeq2 plots saved to: C:/Users/fionn/Newcastle Grammar School/NGS SciX-2025 UoN DMG Cancer Signalling - Bioinformatics and Proteomics/Fionn Git/HSC-SciX-UoN_LS3.33/R Projects/2907/output/visualisation/enhanced_volcano/deseq2
```

### edgeR - Empirical Methods for DE (Negative Binomial Distribution)


``` r
# Create output directories for organized volcano plots
dir.create(here("R Projects", "2907", "output", "visualisation", "enhanced_volcano"), recursive = TRUE, showWarnings = FALSE)
dir.create(here("R Projects", "2907", "output", "visualisation", "enhanced_volcano", "edgeR"), recursive = TRUE, showWarnings = FALSE)

# Define selected genes for labeling
selected_genes <- c("S1pr1", "S1pr2", "S1pr3", "S1pr4", "S1pr5",
                    "Ccr7", "Sell", "Cd69", "Klf2", "Cxcr4", "Cd44", "Itgae", "Itgal")

# Define comparisons
pairwise_comparisons <- list(
  "NAIVE_WB_vs_NAIVE_BM"     = c("Naive_WB", "Naive_BM"),
  "NAIVE_WB_vs_SHAM_WB"      = c("Naive_WB", "Sham_WB"),
  "NAIVE_WB_vs_SHAM_BM"      = c("Naive_WB", "Sham_BM"),
  "NAIVE_WB_vs_UT_WB"        = c("Naive_WB", "UT_WB"),
  "NAIVE_WB_vs_UT_BM"        = c("Naive_WB", "UT_BM"),
  "NAIVE_BM_vs_SHAM_WB"      = c("Naive_BM", "Sham_WB"),
  "NAIVE_BM_vs_SHAM_BM"      = c("Naive_BM", "Sham_BM"),
  "NAIVE_BM_vs_UT_WB"        = c("Naive_BM", "UT_WB"),
  "NAIVE_BM_vs_UT_BM"        = c("Naive_BM", "UT_BM"),
  "SHAM_WB_vs_SHAM_BM"       = c("Sham_WB", "Sham_BM"),
  "SHAM_WB_vs_UT_WB"         = c("Sham_WB", "UT_WB"),
  "SHAM_WB_vs_UT_BM"         = c("Sham_WB", "UT_BM"),
  "SHAM_BM_vs_UT_WB"         = c("Sham_BM", "UT_WB"),
  "SHAM_BM_vs_UT_BM"         = c("Sham_BM", "UT_BM"),
  "UT_WB_vs_UT_BM"           = c("UT_WB", "UT_BM")
)

# Run edgeR analysis and generate volcano plots for each comparison
for (i in names(pairwise_comparisons)) {
  
  group1 <- pairwise_comparisons[[i]][1]
  group2 <- pairwise_comparisons[[i]][2]
  
  cat("\n", rep("=", 50), "\n")
  cat("Processing:", i, "\n")
  cat(rep("=", 50), "\n")
  
  # Run edgeR analysis for this comparison
  deg_T <- run_edgeR_comparison(pseudobulk_counts, sample_info, group1, group2, i)
  
  if (!is.null(deg_T) && nrow(deg_T) > 0) {
    
    # Create volcano plot
    p <- EnhancedVolcano(
      deg_T,
      lab = deg_T$gene,
      x = "avg_log2FC",           # Using renamed column from edgeR results
      y = "p_val_adj",            # Using renamed column from edgeR results
      selectLab = intersect(deg_T$gene, selected_genes),
      pCutoff = 0.05,
      FCcutoff = 0.5,             # Slightly higher FC cutoff for pseudobulk data
      title = paste("edgeR:", i),
      subtitle = paste("Pseudobulk analysis -", group1, "vs", group2),
      caption = paste("Total genes in plot:", nrow(deg_T)),
      pointSize = 3.0,
      labSize = 4.5,
      drawConnectors = TRUE,
      widthConnectors = 0.5,
      colConnectors = "grey50",
      legendPosition = "right",
      legendLabSize = 12,
      legendIconSize = 4.0,
      # Color scheme
      col = c("grey30", "forestgreen", "royalblue", "red2"),
      colAlpha = 0.7,              # Only specify colAlpha once
      # Axis limits
      xlim = c(min(deg_T$avg_log2FC, na.rm = TRUE) - 0.5, 
               max(deg_T$avg_log2FC, na.rm = TRUE) + 0.5),
      ylim = c(0, max(-log10(deg_T$p_val_adj), na.rm = TRUE) + 1),
      # Grid and theme
      gridlines.major = TRUE,
      gridlines.minor = FALSE,
      border = "full",
      borderWidth = 1.0,
      borderColour = "black"
    )
    
    # Add custom theme elements
    p <- p + 
      theme_bw() +
      theme(
        plot.title = element_text(size = 14, face = "bold", hjust = 0.5),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        axis.title = element_text(size = 12, face = "bold"),
        axis.text = element_text(size = 11),
        legend.title = element_text(size = 12, face = "bold"),
        legend.text = element_text(size = 11),
        panel.grid.major = element_line(color = "grey90", size = 0.5),
        panel.border = element_rect(color = "black", fill = NA, size = 1)
      )
    
    # Display the plot
    print(p)
    
    # Save the plot to edgeR subfolder
    ggsave(
      filename = file.path(here("R Projects", "2907", "output", "visualisation", "enhanced_volcano", "edgeR"), 
                          paste0(i, "_edgeR_pseudobulk.svg")),
      plot = p,
      width = 14,
      height = 10,
      dpi = 300,
      bg = "white"
    )
    
    # Also save as PNG for backup
    ggsave(
      filename = file.path(here("R Projects", "2907", "output", "visualisation", "enhanced_volcano", "edgeR"), 
                          paste0(i, "_edgeR_pseudobulk.png")),
      plot = p,
      width = 14,
      height = 10,
      dpi = 300,
      bg = "white"
    )
    
    cat("✓ Volcano plot created and saved for:", i, "\n")
    cat("  - Genes in plot:", nrow(deg_T), "\n")
    cat("  - Significant genes (FDR < 0.05):", sum(deg_T$p_val_adj < 0.05, na.rm = TRUE), "\n")
    cat("  - Upregulated (FC > 0.5, FDR < 0.05):", 
        sum(deg_T$avg_log2FC > 0.5 & deg_T$p_val_adj < 0.05, na.rm = TRUE), "\n")
    cat("  - Downregulated (FC < -0.5, FDR < 0.05):", 
        sum(deg_T$avg_log2FC < -0.5 & deg_T$p_val_adj < 0.05, na.rm = TRUE), "\n")
    
  } else {
    cat("✗ No results generated for:", i, "\n")
    cat("  This may be due to insufficient samples or no detected expression\n")
  }
}
```

```
## 
##  = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = 
## Processing: NAIVE_WB_vs_NAIVE_BM 
## = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = =
```

```
## Error in run_edgeR_comparison(pseudobulk_counts, sample_info, group1, : could not find function "run_edgeR_comparison"
```

``` r
cat("\n" , rep("=", 60), "\n")
```

```
## 
##  = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = =
```

``` r
cat("EDGER VOLCANO PLOT GENERATION COMPLETED!\n")
```

```
## EDGER VOLCANO PLOT GENERATION COMPLETED!
```

``` r
cat(rep("=", 60), "\n")
```

```
## = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = =
```

``` r
cat("All plots saved to:", here("R Projects", "2907", "output", "visualisation", "enhanced_volcano", "edgeR"), "\n")
```

```
## All plots saved to: C:/Users/fionn/Newcastle Grammar School/NGS SciX-2025 UoN DMG Cancer Signalling - Bioinformatics and Proteomics/Fionn Git/HSC-SciX-UoN_LS3.33/R Projects/2907/output/visualisation/enhanced_volcano/edgeR
```

``` r
cat("File formats: SVG (vector) and PNG (raster)\n")
```

```
## File formats: SVG (vector) and PNG (raster)
```

### Student's T Test


``` r
# Define the gene list you want to focus on
selected_genes <- c("S1pr1", "S1pr2", "S1pr3", "S1pr4", "S1pr5", 
                    "Ccr7", "Sell", "Cd69", "Klf2", "Cxcr4", "Cd44", "Itgae", "Itgal")

pairwise_comparisons <- list(
  "NAIVE_WB_vs_NAIVE_BM"     = c("Naive_WB", "Naive_BM"),
  "NAIVE_WB_vs_SHAM_WB"      = c("Naive_WB", "Sham_WB"),
  "NAIVE_WB_vs_SHAM_BM"      = c("Naive_WB", "Sham_BM"),
  "NAIVE_WB_vs_UT_WB"        = c("Naive_WB", "UT_WB"),
  "NAIVE_WB_vs_UT_BM"        = c("Naive_WB", "UT_BM"),
  "NAIVE_BM_vs_SHAM_WB"      = c("Naive_BM", "Sham_WB"),
  "NAIVE_BM_vs_SHAM_BM"      = c("Naive_BM", "Sham_BM"),
  "NAIVE_BM_vs_UT_WB"        = c("Naive_BM", "UT_WB"),
  "NAIVE_BM_vs_UT_BM"        = c("Naive_BM", "UT_BM"),
  "SHAM_WB_vs_SHAM_BM"       = c("Sham_WB", "Sham_BM"),
  "SHAM_WB_vs_UT_WB"         = c("Sham_WB", "UT_WB"),
  "SHAM_WB_vs_UT_BM"         = c("Sham_WB", "UT_BM"),
  "SHAM_BM_vs_UT_WB"         = c("Sham_BM", "UT_WB"),
  "SHAM_BM_vs_UT_BM"         = c("Sham_BM", "UT_BM"),
  "UT_WB_vs_UT_BM"           = c("UT_WB", "UT_BM")
)

# Create an Excel workbook
wb <- createWorkbook()

for (i in names(pairwise_comparisons)) {
  
  deg_T <- FindMarkers(
    seu_T,
    group.by = "sample_name",
    ident.1 = pairwise_comparisons[[i]][1],
    ident.2 = pairwise_comparisons[[i]][2],
    test.use = "t"
  )
  
  addWorksheet(wb, sheetName = i)
  
  # Filter for selected genes and sort by log2FC
  deg_T <- deg_T %>%
    filter(rownames(.) %in% selected_genes) %>%
    arrange(-avg_log2FC)
  
  # Add gene names as a column and reorder
  deg_T$gene <- rownames(deg_T)
  deg_T <- deg_T %>% dplyr::select(gene, dplyr::everything())
  
  # Write to Excel sheet
  writeData(wb, sheet = i, x = deg_T)
}

# Save the workbook
saveWorkbook(
  wb,
  here("R Projects", "2907", "output", "de_analysis", "Find Markers", "DE_t_filtered.xlsx"),
  overwrite = TRUE
)
```

# Receptor Localistion & Surface vs Total Expression

# Ligand–Receptor Interaction Analysis

# Trajectory Analysis/Pseudotime


``` r
cds <- as.cell_data_set(seu_NKT_focused)
cds <- preprocess_cds(cds, num_dim = 50)
cds <- cluster_cells(cds, reduction_method = "UMAP")  # <-- fix here
cds <- reduce_dimension(cds, reduction_method = "UMAP")
cds <- learn_graph(cds)
plot_cells(cds, color_cells_by = "pseudotime")
```

# Visualisation

## Dittoplot

### Dittoplot Vln Plot


``` r
dittoPlot(seu_T, "S1pr1", group.by = "treatment", split.by = "tissue")
```

<img src="2907_files/figure-html/unnamed-chunk-26-1.svg" width="90%" keepaspectratio=true style="display: block; margin: auto;" />

### Ditto Plot - Box Plot


``` r
dittoBoxPlot(seu_T, "S1pr1", group.by = "treatment", split.by = "tissue")
```

<img src="2907_files/figure-html/unnamed-chunk-27-1.svg" width="90%" keepaspectratio=true style="display: block; margin: auto;" />

### Ditto Plot - Ridge Plot


``` r
dittoRidgePlot(seu_T, "S1pr1", group.by = "treatment", split.by = "tissue")
```

```
## Picking joint bandwidth of 0.188
```

```
## Picking joint bandwidth of 0.205
```

<img src="2907_files/figure-html/unnamed-chunk-28-1.svg" width="90%" keepaspectratio=true style="display: block; margin: auto;" />

## Enhanced Volcano Plots

## Deseq

See under DE results

## Edge R

See under DE results

## MAST

see under DE results

## Nebulosa

### Seurat T Cluster


``` r
Nebulosa::plot_density(seu_T, "S1pr1") + 
    facet_wrap(.~seu_T$sample_name, ncol = 3) + theme_void()
```

```
## Warning: The `slot` argument of `FetchData()` is deprecated as of SeuratObject 5.0.0.
## ℹ Please use the `layer` argument instead.
## ℹ The deprecated feature was likely used in the Nebulosa package.
##   Please report the issue at
##   <https://github.com/powellgenomicslab/Nebulosa/issues>.
## This warning is displayed once every 8 hours.
## Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
## generated.
```

<img src="2907_files/figure-html/unnamed-chunk-29-1.svg" width="90%" keepaspectratio=true style="display: block; margin: auto;" />

``` r
Nebulosa::plot_density(seu_T, "S1pr2") + 
    facet_wrap(.~seu_T$sample_name, ncol = 3) + theme_void()
```

<img src="2907_files/figure-html/unnamed-chunk-29-2.svg" width="90%" keepaspectratio=true style="display: block; margin: auto;" />

``` r
Nebulosa::plot_density(seu_T, "S1pr3") + 
    facet_wrap(.~seu_T$sample_name, ncol = 3) + theme_void()
```

<img src="2907_files/figure-html/unnamed-chunk-29-3.svg" width="90%" keepaspectratio=true style="display: block; margin: auto;" />

``` r
Nebulosa::plot_density(seu_T, "S1pr4") + 
    facet_wrap(.~seu_T$sample_name, ncol = 3) + theme_void()
```

<img src="2907_files/figure-html/unnamed-chunk-29-4.svg" width="90%" keepaspectratio=true style="display: block; margin: auto;" />

``` r
Nebulosa::plot_density(seu_T, "S1pr5") + 
    facet_wrap(.~seu_T$sample_name, ncol = 3) + theme_void()
```

<img src="2907_files/figure-html/unnamed-chunk-29-5.svg" width="90%" keepaspectratio=true style="display: block; margin: auto;" />
