```{r}
knitr::opts_chunk$set(
    error     = TRUE,     # Ensures that error messages are being shown in the output
    fig.align = "center", # Ensures that all figures are aligned to the centre in the output
    message   = TRUE,     # Display messages generated by R
    warning   = TRUE,     # Display warnings generated by R
    autodep   = TRUE,     # Automatically detects dependencies between chunks and reruns dependent chunks when needed
    cache     = FALSE,    # Disables caching, meaning every time the document is knit, all chunks are re-executed
    results   = "markup", # Displays the output as normal formatted text
    echo      = TRUE     # Ensures that the R code itself is displayed in the document
    #dev       = "svg",    # Graphics output will be in Scalable Vector Graphics (SVG) format
    #out.width = "90%",    # Sets the width of the output image or plot to 90% of the available width 
  #out.extra = "keepaspectratio=true" # Adds extra options to the output, in this case, preserving the aspect ratio of the image or plot
)
```

```{r}
# Maximum allowable size for globals in bytes that can be exported to the parallel workers
mib <- 9000*1024^2
options(future.globals.maxSize = mib)  # 1.5 GiB
```

```{r}
library(here)
```

# Set Up Working Environment

## Load Packages & Functions

```{r}
source(here("Custom R Functions and Scripts", "init_packages.R"))
```

# Configuration

## Custom Colours

```{r}
# Custom colors
man_cols <- c("#86b0cc",  "#f3e65d", "#d5c1e7", "#eeb84c", "#82c39e", "#525252","#4d9f6b", "#b3939e", "#e76031", "#e9944b")
names(man_cols) <- c("B_cells", "NK_cells", "monocytes", "T_cells", "neutrophils", "megakaryocytes", "pDCs",
                     "plasma_cells", "progenitor_cells", "NKT_cells")
```

## Output Folders

### Primary

```{r}
dir.create(here("R Projects", "2107", "output"))
```

### DGE

```{r}
dir.create(here("R Projects", "2107", "output", "de_analysis"))
```

# Import & Pre-processing of Data

## Load RDS file

### Load Original RDS

Importing PhD Clara Savary's RDS & Initialising the object as a Seurat Object

```{r}

seu <- readRDS(here("Data",          "seu_NKT_blood_mouse_PPK_ONC_harmony_integrated_filtered_v2_20231_20236_annot.rds"))
```

### Subset

Initial Subset to isolate T Cells, Natural Killer T Cells and Natural Killer Cells and label

```{r}
seu_NKT <- subset(seu, subset = cell_types %in% c("T_cells", "NK_cells", "NKT_cells"))
```

### Subset Dataset Again

Second Subset -\> subsetting to remove treatment conditions involving theraputics as this is outside the bounds of the scope of the investigation.

Including:

-   DMG Mice (represented as untreated or UT)

-   Control: No Cancer (Treatment and Disease Naive Group)

-   Surgery Control (SHAM - Injected with Saline - to make sure that the stress of surgery isn't a factor and can be compared between Naive, UT and Sham)

Excluding:

-   ONC201 - Dordaviprone -\> DRD2 antagonist and ClpP agonist

    -   <https://doi.org/10.1158/0008-5472.CAN-23-0186>

    -   <https://doi.org/10.1158/2159-8290.CD-23-0131>

-   Dex - Dexmethasone -\> Corticosteroid

-   Dex & ONC201 Combo

```{r}
seu_NKT_focused <- subset(seu_NKT, subset = treatment %in% c("UT", "NAIVE", "SHAM"))
```

For later scripting - load seu_NKT_focused

```{r}
seu_NKT_focused <- readRDS(here("Data", "seu_NKT_focused.RDS"))
```

# Differential Gene Expression Analysis

## T Cells

```{r}
seu_T <- subset(seu_NKT_focused, subset = T_clusters %in% "T_cells")
```

### Wilcoxon signed-rank test

```{r}

# Define the gene list you want to focus on
selected_genes <- c("S1pr1", "S1pr2", "S1pr3", "S1pr4", "S1pr5", 
                    "Ccr7", "Sell", "Cd69", "Klf2", "Cxcr4", "Cd44", "Itgae", "Itgal")

pairwise_comparisons <- list(
  "NAIVE_WB_vs_NAIVE_BM"     = c("Naive_WB", "Naive_BM"),
  "NAIVE_WB_vs_SHAM_WB"      = c("Naive_WB", "Sham_WB"),
  "NAIVE_WB_vs_SHAM_BM"      = c("Naive_WB", "Sham_BM"),
  "NAIVE_WB_vs_UT_WB"        = c("Naive_WB", "UT_WB"),
  "NAIVE_WB_vs_UT_BM"        = c("Naive_WB", "UT_BM"),
  "NAIVE_BM_vs_SHAM_WB"      = c("Naive_BM", "Sham_WB"),
  "NAIVE_BM_vs_SHAM_BM"      = c("Naive_BM", "Sham_BM"),
  "NAIVE_BM_vs_UT_WB"        = c("Naive_BM", "UT_WB"),
  "NAIVE_BM_vs_UT_BM"        = c("Naive_BM", "UT_BM"),
  "SHAM_WB_vs_SHAM_BM"       = c("Sham_WB", "Sham_BM"),
  "SHAM_WB_vs_UT_WB"         = c("Sham_WB", "UT_WB"),
  "SHAM_WB_vs_UT_BM"         = c("Sham_WB", "UT_BM"),
  "SHAM_BM_vs_UT_WB"         = c("Sham_BM", "UT_WB"),
  "SHAM_BM_vs_UT_BM"         = c("Sham_BM", "UT_BM"),
  "UT_WB_vs_UT_BM"           = c("UT_WB", "UT_BM")
)

wb <- createWorkbook()

for (i in names(pairwise_comparisons)) {
  
  deg_T <- wilcoxauc(seu_T, group_by = "sample_name", groups_use = pairwise_comparisons[[i]])
  
  addWorksheet(wb, sheetName = i)
  
  deg_T <- deg_T %>%
    filter(group == pairwise_comparisons[[i]][[1]]) %>%
    filter(feature %in% selected_genes) %>%   # <- Filter only your selected genes
    arrange(group, -logFC)
  
  writeData(wb, sheet = i, x = deg_T)
}

saveWorkbook(wb, here("R Projects", "2107", "output", "de_analysis", "DE_presto_filtered.xlsx"), overwrite = TRUE)

```

### deseq2

```{r}
# Define the gene list you want to focus on
selected_genes <- c("S1pr1", "S1pr2", "S1pr3", "S1pr4", "S1pr5", 
                    "Ccr7", "Sell", "Cd69", "Klf2", "Cxcr4", "Cd44", "Itgae", "Itgal")

pairwise_comparisons <- list(
  "NAIVE_WB_vs_NAIVE_BM"     = c("Naive_WB", "Naive_BM"),
  "NAIVE_WB_vs_SHAM_WB"      = c("Naive_WB", "Sham_WB"),
  "NAIVE_WB_vs_SHAM_BM"      = c("Naive_WB", "Sham_BM"),
  "NAIVE_WB_vs_UT_WB"        = c("Naive_WB", "UT_WB"),
  "NAIVE_WB_vs_UT_BM"        = c("Naive_WB", "UT_BM"),
  "NAIVE_BM_vs_SHAM_WB"      = c("Naive_BM", "Sham_WB"),
  "NAIVE_BM_vs_SHAM_BM"      = c("Naive_BM", "Sham_BM"),
  "NAIVE_BM_vs_UT_WB"        = c("Naive_BM", "UT_WB"),
  "NAIVE_BM_vs_UT_BM"        = c("Naive_BM", "UT_BM"),
  "SHAM_WB_vs_SHAM_BM"       = c("Sham_WB", "Sham_BM"),
  "SHAM_WB_vs_UT_WB"         = c("Sham_WB", "UT_WB"),
  "SHAM_WB_vs_UT_BM"         = c("Sham_WB", "UT_BM"),
  "SHAM_BM_vs_UT_WB"         = c("Sham_BM", "UT_WB"),
  "SHAM_BM_vs_UT_BM"         = c("Sham_BM", "UT_BM"),
  "UT_WB_vs_UT_BM"           = c("UT_WB", "UT_BM")
)

wb <- createWorkbook()

for (i in names(pairwise_comparisons)) {
  
  # Use pseudobulk approach for DESeq2 or use a different test
  # Option 1: Use negative binomial test instead (more appropriate for scRNA-seq)
  deg_T <- FindMarkers(seu_T, 
                       ident.1 = pairwise_comparisons[[i]][[1]], 
                       ident.2 = pairwise_comparisons[[i]][[2]], 
                       group.by = "sample_name", 
                       test.use = "negbinom",
                       slot = "counts")  # Use raw counts
  
  addWorksheet(wb, sheetName = i)
  
  deg_T <- deg_T %>%
    rownames_to_column("feature") %>%
    mutate(group = pairwise_comparisons[[i]][[1]]) %>%
    rename(logFC = avg_log2FC) %>%
    filter(feature %in% selected_genes) %>%   # <- Filter only your selected genes
    arrange(group, -logFC)
  
  writeData(wb, sheet = i, x = deg_T)
}

saveWorkbook(wb, here("R Projects", "2107", "output", "de_analysis", "DE_deseq2_filtered.xlsx"), overwrite = TRUE)
```

### edgeR - Empirical Methods for DE (Negative Binomial Distribution)

```{r}

```

###  

### MAST

```{r}
# Define the gene list you want to focus on
selected_genes <- c("S1pr1", "S1pr2", "S1pr3", "S1pr4", "S1pr5", 
                    "Ccr7", "Sell", "Cd69", "Klf2", "Cxcr4", "Cd44", "Itgae", "Itgal")

pairwise_comparisons <- list(
  "NAIVE_WB_vs_NAIVE_BM"     = c("Naive_WB", "Naive_BM"),
  "NAIVE_WB_vs_SHAM_WB"      = c("Naive_WB", "Sham_WB"),
  "NAIVE_WB_vs_SHAM_BM"      = c("Naive_WB", "Sham_BM"),
  "NAIVE_WB_vs_UT_WB"        = c("Naive_WB", "UT_WB"),
  "NAIVE_WB_vs_UT_BM"        = c("Naive_WB", "UT_BM"),
  "NAIVE_BM_vs_SHAM_WB"      = c("Naive_BM", "Sham_WB"),
  "NAIVE_BM_vs_SHAM_BM"      = c("Naive_BM", "Sham_BM"),
  "NAIVE_BM_vs_UT_WB"        = c("Naive_BM", "UT_WB"),
  "NAIVE_BM_vs_UT_BM"        = c("Naive_BM", "UT_BM"),
  "SHAM_WB_vs_SHAM_BM"       = c("Sham_WB", "Sham_BM"),
  "SHAM_WB_vs_UT_WB"         = c("Sham_WB", "UT_WB"),
  "SHAM_WB_vs_UT_BM"         = c("Sham_WB", "UT_BM"),
  "SHAM_BM_vs_UT_WB"         = c("Sham_BM", "UT_WB"),
  "SHAM_BM_vs_UT_BM"         = c("Sham_BM", "UT_BM"),
  "UT_WB_vs_UT_BM"           = c("UT_WB", "UT_BM")
)

library(MAST)
library(openxlsx)
library(dplyr)
library(here)
library(Seurat)

wb_mast <- createWorkbook()

for (i in names(pairwise_comparisons)) {
  
  cells_use <- WhichCells(seu_T, expression = sample_name %in% pairwise_comparisons[[i]])
  seu_sub <- subset(seu_T, cells = cells_use)
  
  expr_mat <- GetAssayData(seu_sub, assay = "RNA", slot = "counts")
  cdr <- colSums(expr_mat > 0)  # Cellular detection rate
  
  sca <- FromMatrix(exprsArray = expr_mat,
                    cData = data.frame(wellKey = colnames(expr_mat),
                                       sample_name = seu_sub@meta.data$sample_name,
                                       cdr = cdr))
  
  sca$cData$condition <- factor(sca$cData$sample_name, levels = pairwise_comparisons[[i]])
  
  zlmCond <- zlm(~ condition + cdr, sca)
  summaryCond <- summary(zlmCond, doLRT = "condition2")
  summaryDt <- summaryCond$datatable
  
  deg_T <- summaryDt %>%
    filter(component == "H" & contrast == "condition2") %>%
   

```
