---
title: "R Notebook"
output: html_notebook
---

```{r setup, include=FALSE}
knitr::opts_knit$set(root.dir = rprojroot::find_rstudio_root_file())
knitr::opts_chunk$set(
    error     = TRUE,     # Ensures that error messages are being shown in the output
    fig.align = "center", # Ensures that all figures are aligned to the centre in the output
    message   = TRUE,     # Display messages generated by R
    warning   = TRUE,     # Display warnings generated by R
    autodep   = TRUE,     # Automatically detects dependencies between chunks and reruns dependent chunks when needed
    cache     = FALSE,    # Disables caching, meaning every time the document is knit, all chunks are re-executed
    results   = "markup", # Displays the output as normal formatted text
    echo      = TRUE     # Ensures that the R code itself is displayed in the document
    #dev       = "svg",    # Graphics output will be in Scalable Vector Graphics (SVG) format
    #out.width = "90%",    # Sets the width of the output image or plot to 90% of the available width 
  #out.extra = "keepaspectratio=true" # Adds extra options to the output, in this case, preserving the aspect ratio of the image or plot
)
```

# Packages

## Install

```{r}
source("C:/Users/fionn/Newcastle Grammar School/NGS SciX-2025 UoN DMG Cancer Signalling - Bioinformatics and Proteomics/Fionn Git/HSC-SciX-UoN_LS3.33/Custom R Functions and Scripts/install_packages.R")
```

```{r}
source("C:/Users/fionn/Newcastle Grammar School/NGS SciX-2025 UoN DMG Cancer Signalling - Bioinformatics and Proteomics/Fionn Git/HSC-SciX-UoN_LS3.33/Custom R Functions and Scripts/init_packages.R")
```

# Clara's Colours

```{r}
man_cols <- c("#86b0cc",  "#f3e65d", "#d5c1e7", "#eeb84c", "#82c39e", "#525252","#4d9f6b", "#b3939e", "#e76031", "#e9944b")
names(man_cols) <- c("B_cells", "NK_cells", "monocytes", "T_cells", "neutrophils", "megakaryocytes", "pDCs",
                     "plasma_cells", "progenitor_cells", "NKT_cells")
```

# Import Data

## RDS Files

### Load Clara's Exported RDS

```{r}
seu <- readRDS("./Data/seu_NKT_blood_mouse_PPK_ONC_harmony_integrated_filtered_v2_20231_20236_annot.rds")
```

### Subset Seurat Object to Isolate NK & T Cells

btw because Clara already did this and gave this to me as the NKT subset seu & seu_NKT are technically the same

```{r}
seu_NKT <- subset(seu, subset = cell_types %in% c("T_cells", "NK_cells", "NKT_cells"))
```

### Subset *"seu_NKT"* to isolate conditions to be investigated -\> No pharmaco

```{r}
seu_NKT_focused <- subset(seu_NKT, subset = treatment %in% c("Naive", "Sham", "UT"))
```
