---
output_dir: R Projects/2107claude/output/rmd_outs
output:
  html_document:
    toc: true
    toc_depth: 6
    toc_float: true
    code_folding: hide
    keep_md: true
    number_sections: true
  pdf_document:
    toc: true
    toc_depth: '6'
---

```{r}
library(here) 

# Set global knitr chunk options
knitr::opts_chunk$set(
    error     = TRUE,        # Ensures that error messages are being shown in the output
    fig.align = "center",    # Ensures that all figures are aligned to the centre in the output
    message   = TRUE,       # <--- Recommend FALSE for cleaner reports, set to TRUE for development
    warning   = TRUE,       # <--- Recommend FALSE for cleaner reports, set to TRUE for development
    autodep   = TRUE,        # Automatically detects dependencies for caching
    cache     = TRUE,        # <--- IMPORTANT: Enable caching globally here!
    results   = "markup",    # Displays the output as normal formatted text
    echo      = TRUE,        # Ensures that the R code itself is displayed in the document
    dev       = "svg",       # Graphics output will be in Scalable Vector Graphics (SVG) format
    out.width = "90%",       # Sets the width of the output image or plot to 90% of the available width 
    out.extra = "keepaspectratio=true" # Adds extra options to the output, preserving aspect ratio
)

# Set global option for future package (important for large parallel computations like Seurat/MAST)
mib <- 9000*1024^2 # 9GB in bytes
options(future.globals.maxSize = mib) 
```

```{r}
library(here)
```

# Set Up Working Environment

## Set Seed

```{r}
seed <- "210825"
set.seed(seed)
message("seed has been set to 210825 ")
```

## Load Packages & Functions

```{r}
source(here("Custom R Functions and Scripts", "init_packages.R"))
```

# Configuration

## Custom Colours

```{r}
# Custom colors
man_cols <- c("#86b0cc",  "#f3e65d", "#d5c1e7", "#eeb84c", "#82c39e", "#525252","#4d9f6b", "#b3939e", "#e76031", "#e9944b")
names(man_cols) <- c("B_cells", "NK_cells", "monocytes", "T_cells", "neutrophils", "megakaryocytes", "pDCs",
                     "plasma_cells", "progenitor_cells", "NKT_cells")
```

## Output Folders

### Primary

```{r}
dir.create(here("R Projects", "2107claude", "output"), recursive = TRUE, showWarnings = FALSE)
```

### DGE

```{r}
dir.create(here("R Projects", "2107claude", "output", "de_analysis", "Find Markers"), recursive = TRUE, showWarnings = FALSE)
dir.create(here("R Projects", "2107claude", "output", "de_analysis", "Presto"), recursive = TRUE, showWarnings = FALSE)
dir.create(here("R Projects", "2107claude", "output", "de_analysis", "t.test"), recursive = TRUE, showWarnings = FALSE)
```

### Visualisation

```{r}
dir.create(here("R Projects", "2107claude", "output", "visualisation", "box plots"), recursive = TRUE, showWarnings = FALSE)
dir.create(here("R Projects", "2107claude", "output", "visualisation", "complex heatmaps"), recursive = TRUE, showWarnings = FALSE)
dir.create(here("R Projects", "2107claude", "output", "visualisation", "nebulosa"), recursive = TRUE, showWarnings = FALSE)
dir.create(here("R Projects", "2107claude", "output", "visualisation", "violin plots"), recursive = TRUE, showWarnings = FALSE)
dir.create(here("R Projects", "2107claude", "output", "visualisation", "volcano"), recursive = TRUE, showWarnings = FALSE)
```

# Import & Pre-processing of Data

## Load RDS file

### Load Original RDS

Importing PhD Clara Savary's RDS & Initialising the object as a Seurat Object

```{r, eval=FALSE}

seu <- readRDS(here("Data",          "seu_NKT_blood_mouse_PPK_ONC_harmony_integrated_filtered_v2_20231_20236_annot.rds"))
```

### Subset

Initial Subset to isolate T Cells, Natural Killer T Cells and Natural Killer Cells and label

```{r, eval=FALSE}
seu_NKT <- subset(seu, subset = cell_types %in% c("T_cells", "NK_cells", "NKT_cells"))
```

### Subset Dataset Again

Second Subset -\> subsetting to remove treatment conditions involving theraputics as this is outside the bounds of the scope of the investigation.

Including:

-   DMG Mice (represented as untreated or UT)

-   Control: No Cancer (Treatment and Disease Naive Group)

-   Surgery Control (SHAM - Injected with Saline - to make sure that the stress of surgery isn't a factor and can be compared between Naive, UT and Sham)

Excluding:

-   ONC201 - Dordaviprone -\> DRD2 antagonist and ClpP agonist

    -   <https://doi.org/10.1158/0008-5472.CAN-23-0186>

    -   <https://doi.org/10.1158/2159-8290.CD-23-0131>

-   Dex - Dexmethasone -\> Corticosteroid

-   Dex & ONC201 Combo

```{r, eval=FALSE}
seu_NKT_focused <- subset(seu_NKT, subset = treatment %in% c("UT", "NAIVE", "SHAM"))
```

For later scripting - load seu_NKT_focused

#### Export

Use either base or BioCgenerics

##### BioCgenerics

```{r,eval=FALSE}
BiocGenerics::saveRDS(seu_NKT_focused, file = here("Data", "seu_NKT_focused.rds"))
```

OR

##### Base

```{r, , eval=FALSE}
base::saveRDS(seu_NKT_focused, file = here("Data", "seu_NKT_focused.rds") )
```

#### Load rds

```{r, cache=FALSE}
seu_NKT_focused <- readRDS(here("Data", "seu_NKT_focused.rds"))
```

# Differential Gene Expression Analysis

## T Cells

```{r, cache=FALSE}
seu_T <- subset(seu_NKT_focused, subset = T_clusters %in% "T_cells")
```

### Presto

#### Wilcoxon signed-rank test

```{r, cache=TRUE}
# Define the gene list you want to focus on
selected_genes <- c("S1pr1", "S1pr2", "S1pr3", "S1pr4", "S1pr5", 
                    "Ccr7", "Sell", "Cd69", "Klf2", "Cxcr4", "Cd44", "Itgae", "Itgal")

pairwise_comparisons <- list(
  "NAIVE_WB_vs_NAIVE_BM"     = c("Naive_WB", "Naive_BM"),
  "NAIVE_WB_vs_SHAM_WB"      = c("Naive_WB", "Sham_WB"),
  "NAIVE_WB_vs_SHAM_BM"      = c("Naive_WB", "Sham_BM"),
  "NAIVE_WB_vs_UT_WB"        = c("Naive_WB", "UT_WB"),
  "NAIVE_WB_vs_UT_BM"        = c("Naive_WB", "UT_BM"),
  "NAIVE_BM_vs_SHAM_WB"      = c("Naive_BM", "Sham_WB"),
  "NAIVE_BM_vs_SHAM_BM"      = c("Naive_BM", "Sham_BM"),
  "NAIVE_BM_vs_UT_WB"        = c("Naive_BM", "UT_WB"),
  "NAIVE_BM_vs_UT_BM"        = c("Naive_BM", "UT_BM"),
  "SHAM_WB_vs_SHAM_BM"       = c("Sham_WB", "Sham_BM"),
  "SHAM_WB_vs_UT_WB"         = c("Sham_WB", "UT_WB"),
  "SHAM_WB_vs_UT_BM"         = c("Sham_WB", "UT_BM"),
  "SHAM_BM_vs_UT_WB"         = c("Sham_BM", "UT_WB"),
  "SHAM_BM_vs_UT_BM"         = c("Sham_BM", "UT_BM"),
  "UT_WB_vs_UT_BM"           = c("UT_WB", "UT_BM")
)

wb <- createWorkbook()

for (i in names(pairwise_comparisons)) {
  
  deg_T <- wilcoxauc(seu_T, group_by = "sample_name", groups_use = pairwise_comparisons[[i]])
  
  addWorksheet(wb, sheetName = i)
  
  deg_T <- deg_T %>%
    filter(group == pairwise_comparisons[[i]][[1]]) %>%
    filter(feature %in% selected_genes) %>%   # <- Filter only your selected genes
    arrange(group, -logFC)
  
  writeData(wb, sheet = i, x = deg_T)
}

saveWorkbook(wb, here("R Projects", "2107claude", "output", "de_analysis", "Presto", "DE_presto-wilcoxauc_filtered.xlsx"), overwrite = TRUE)

```

### Find Markers

#### Wilcox Presto

```{r, eval=TRUE, cache=TRUE}
# Define the gene list you want to focus on
selected_genes <- c("S1pr1", "S1pr2", "S1pr3", "S1pr4", "S1pr5", 
                    "Ccr7", "Sell", "Cd69", "Klf2", "Cxcr4", "Cd44", "Itgae", "Itgal")

pairwise_comparisons <- list(
  "NAIVE_WB_vs_NAIVE_BM"     = c("Naive_WB", "Naive_BM"),
  "NAIVE_WB_vs_SHAM_WB"      = c("Naive_WB", "Sham_WB"),
  "NAIVE_WB_vs_SHAM_BM"      = c("Naive_WB", "Sham_BM"),
  "NAIVE_WB_vs_UT_WB"        = c("Naive_WB", "UT_WB"),
  "NAIVE_WB_vs_UT_BM"        = c("Naive_WB", "UT_BM"),
  "NAIVE_BM_vs_SHAM_WB"      = c("Naive_BM", "Sham_WB"),
  "NAIVE_BM_vs_SHAM_BM"      = c("Naive_BM", "Sham_BM"),
  "NAIVE_BM_vs_UT_WB"        = c("Naive_BM", "UT_WB"),
  "NAIVE_BM_vs_UT_BM"        = c("Naive_BM", "UT_BM"),
  "SHAM_WB_vs_SHAM_BM"       = c("Sham_WB", "Sham_BM"),
  "SHAM_WB_vs_UT_WB"         = c("Sham_WB", "UT_WB"),
  "SHAM_WB_vs_UT_BM"         = c("Sham_WB", "UT_BM"),
  "SHAM_BM_vs_UT_WB"         = c("Sham_BM", "UT_WB"),
  "SHAM_BM_vs_UT_BM"         = c("Sham_BM", "UT_BM"),
  "UT_WB_vs_UT_BM"           = c("UT_WB", "UT_BM")
)

wb <- createWorkbook()

for (i in names(pairwise_comparisons)) {
  
  deg_T <- FindMarkers(seu_T, group.by = "sample_name", ident.1 = pairwise_comparisons[[i]][1], ident.2 = pairwise_comparisons[[i]][2], test.use = "wilcox")
  
  addWorksheet(wb, sheetName = i)
  
  deg_T <- deg_T %>%
    filter(rownames(.) %in% selected_genes) %>%   # <- Filter only your selected genes
    arrange(-avg_log2FC)
  
  # Add gene names as a column
  deg_T$gene <- rownames(deg_T)
  deg_T <- deg_T %>% dplyr::select(gene, dplyr::everything())
  
  writeData(wb, sheet = i, x = deg_T)
}

saveWorkbook(wb, here("R Projects", "2107claude", "output", "de_analysis", "Find Markers", "DE_wilcox_filtered.xlsx"), overwrite = TRUE)
```

#### Wilcoxon Limma

```{r, cache=TRUE}
# Define the gene list you want to focus on
selected_genes <- c("S1pr1", "S1pr2", "S1pr3", "S1pr4", "S1pr5", 
                    "Ccr7", "Sell", "Cd69", "Klf2", "Cxcr4", "Cd44", "Itgae", "Itgal")

pairwise_comparisons <- list(
  "NAIVE_WB_vs_NAIVE_BM"     = c("Naive_WB", "Naive_BM"),
  "NAIVE_WB_vs_SHAM_WB"      = c("Naive_WB", "Sham_WB"),
  "NAIVE_WB_vs_SHAM_BM"      = c("Naive_WB", "Sham_BM"),
  "NAIVE_WB_vs_UT_WB"        = c("Naive_WB", "UT_WB"),
  "NAIVE_WB_vs_UT_BM"        = c("Naive_WB", "UT_BM"),
  "NAIVE_BM_vs_SHAM_WB"      = c("Naive_BM", "Sham_WB"),
  "NAIVE_BM_vs_SHAM_BM"      = c("Naive_BM", "Sham_BM"),
  "NAIVE_BM_vs_UT_WB"        = c("Naive_BM", "UT_WB"),
  "NAIVE_BM_vs_UT_BM"        = c("Naive_BM", "UT_BM"),
  "SHAM_WB_vs_SHAM_BM"       = c("Sham_WB", "Sham_BM"),
  "SHAM_WB_vs_UT_WB"         = c("Sham_WB", "UT_WB"),
  "SHAM_WB_vs_UT_BM"         = c("Sham_WB", "UT_BM"),
  "SHAM_BM_vs_UT_WB"         = c("Sham_BM", "UT_WB"),
  "SHAM_BM_vs_UT_BM"         = c("Sham_BM", "UT_BM"),
  "UT_WB_vs_UT_BM"           = c("UT_WB", "UT_BM")
)

wb <- createWorkbook()

for (i in names(pairwise_comparisons)) {
  
  deg_T <- FindMarkers(seu_T, group.by = "sample_name", ident.1 = pairwise_comparisons[[i]][1], ident.2 = pairwise_comparisons[[i]][2], test.use = "wilcox_limma")
  
  addWorksheet(wb, sheetName = i)
  
  deg_T <- deg_T %>%
    filter(rownames(.) %in% selected_genes) %>%   # <- Filter only your selected genes
    arrange(-avg_log2FC)
  
  # Add gene names as a column
  deg_T$gene <- rownames(deg_T)
  deg_T <- deg_T %>% dplyr::select(gene, dplyr::everything())
  
  writeData(wb, sheet = i, x = deg_T)
}

saveWorkbook(wb, here("R Projects", "2107claude", "output", "de_analysis", "Find Markers", "DE_wilcox-limma_filtered.xlsx"), overwrite = TRUE)

```

#### deseq2

```{r,eval=FALSE, cache=TRUE}
# Define the gene list you want to focus on
selected_genes <- c("S1pr1", "S1pr2", "S1pr3", "S1pr4", "S1pr5", 
                    "Ccr7", "Sell", "Cd69", "Klf2", "Cxcr4", "Cd44", "Itgae", "Itgal")

pairwise_comparisons <- list(
  "NAIVE_WB_vs_NAIVE_BM"     = c("Naive_WB", "Naive_BM"),
  "NAIVE_WB_vs_SHAM_WB"      = c("Naive_WB", "Sham_WB"),
  "NAIVE_WB_vs_SHAM_BM"      = c("Naive_WB", "Sham_BM"),
  "NAIVE_WB_vs_UT_WB"        = c("Naive_WB", "UT_WB"),
  "NAIVE_WB_vs_UT_BM"        = c("Naive_WB", "UT_BM"),
  "NAIVE_BM_vs_SHAM_WB"      = c("Naive_BM", "Sham_WB"),
  "NAIVE_BM_vs_SHAM_BM"      = c("Naive_BM", "Sham_BM"),
  "NAIVE_BM_vs_UT_WB"        = c("Naive_BM", "UT_WB"),
  "NAIVE_BM_vs_UT_BM"        = c("Naive_BM", "UT_BM"),
  "SHAM_WB_vs_SHAM_BM"       = c("Sham_WB", "Sham_BM"),
  "SHAM_WB_vs_UT_WB"         = c("Sham_WB", "UT_WB"),
  "SHAM_WB_vs_UT_BM"         = c("Sham_WB", "UT_BM"),
  "SHAM_BM_vs_UT_WB"         = c("Sham_BM", "UT_WB"),
  "SHAM_BM_vs_UT_BM"         = c("Sham_BM", "UT_BM"),
  "UT_WB_vs_UT_BM"           = c("UT_WB", "UT_BM")
)

wb <- createWorkbook()

for (i in names(pairwise_comparisons)) {
  
  deg_T <- FindMarkers(seu_T, group.by = "sample_name", ident.1 = pairwise_comparisons[[i]][1], ident.2 = pairwise_comparisons[[i]][2], test.use = "DESeq2")
  
  addWorksheet(wb, sheetName = i)
  
  deg_T <- deg_T %>%
    filter(rownames(.) %in% selected_genes) %>%   # <- Filter only your selected genes
    arrange(-avg_log2FC)
  
  # Add gene names as a column
  deg_T$gene <- rownames(deg_T)
  deg_T <- deg_T %>% dplyr::select(gene, dplyr::everything())
  
  writeData(wb, sheet = i, x = deg_T)
}

saveWorkbook(wb, here("R Projects", "2107claude", "output", "de_analysis", "Find Markers", "DE_deseq2_filtered.xlsx"), overwrite = TRUE)

```

#### edgeR - Empirical Methods for DE (Negative Binomial Distribution)

```{r, eval=FALSE, cache=TRUE}
selected_genes <- c("S1pr1", "S1pr2", "S1pr3", "S1pr4", "S1pr5", 
                    "Ccr7", "Sell", "Cd69", "Klf2", "Cxcr4", "Cd44", "Itgae", "Itgal")

pairwise_comparisons <- list(
  "NAIVE_WB_vs_NAIVE_BM"     = c("Naive_WB", "Naive_BM"),
  "NAIVE_WB_vs_SHAM_WB"      = c("Naive_WB", "Sham_WB"),
  "NAIVE_WB_vs_SHAM_BM"      = c("Naive_WB", "Sham_BM"),
  "NAIVE_WB_vs_UT_WB"        = c("Naive_WB", "UT_WB"),
  "NAIVE_WB_vs_UT_BM"        = c("Naive_WB", "UT_BM"),
  "NAIVE_BM_vs_SHAM_WB"      = c("Naive_BM", "Sham_WB"),
  "NAIVE_BM_vs_SHAM_BM"      = c("Naive_BM", "Sham_BM"),
  "NAIVE_BM_vs_UT_WB"        = c("Naive_BM", "UT_WB"),
  "NAIVE_BM_vs_UT_BM"        = c("Naive_BM", "UT_BM"),
  "SHAM_WB_vs_SHAM_BM"       = c("Sham_WB", "Sham_BM"),
  "SHAM_WB_vs_UT_WB"         = c("Sham_WB", "UT_WB"),
  "SHAM_WB_vs_UT_BM"         = c("Sham_WB", "UT_BM"),
  "SHAM_BM_vs_UT_WB"         = c("Sham_BM", "UT_WB"),
  "SHAM_BM_vs_UT_BM"         = c("Sham_BM", "UT_BM"),
  "UT_WB_vs_UT_BM"           = c("UT_WB", "UT_BM")
)

wb <- createWorkbook()

for (i in names(pairwise_comparisons)) {
  
  deg_T <- FindMarkers(seu_T, group.by = "sample_name", ident.1 = pairwise_comparisons[[i]][1], ident.2 = pairwise_comparisons[[i]][2], test.use = "edgeR")
  
  addWorksheet(wb, sheetName = i)
  
  deg_T <- deg_T %>%
    filter(rownames(.) %in% selected_genes) %>%   # <- Filter only your selected genes
    arrange(-avg_log2FC)
  
  # Add gene names as a column
  deg_T$gene <- rownames(deg_T)
  deg_T <- deg_T %>% dplyr::select(gene, dplyr::everything())
  
  writeData(wb, sheet = i, x = deg_T)
}

saveWorkbook(wb, here("R Projects", "2107claude", "output", "de_analysis", "Find Markers", "DE_edgeR_filtered.xlsx"), overwrite = TRUE)


```

#### MAST

```{r, eval=TRUE, cache=TRUE}
# Define the gene list you want to focus on
selected_genes <- c("S1pr1", "S1pr2", "S1pr3", "S1pr4", "S1pr5", 
                    "Ccr7", "Sell", "Cd69", "Klf2", "Cxcr4", "Cd44", "Itgae", "Itgal")

pairwise_comparisons <- list(
  "NAIVE_WB_vs_NAIVE_BM"     = c("Naive_WB", "Naive_BM"),
  "NAIVE_WB_vs_SHAM_WB"      = c("Naive_WB", "Sham_WB"),
  "NAIVE_WB_vs_SHAM_BM"      = c("Naive_WB", "Sham_BM"),
  "NAIVE_WB_vs_UT_WB"        = c("Naive_WB", "UT_WB"),
  "NAIVE_WB_vs_UT_BM"        = c("Naive_WB", "UT_BM"),
  "NAIVE_BM_vs_SHAM_WB"      = c("Naive_BM", "Sham_WB"),
  "NAIVE_BM_vs_SHAM_BM"      = c("Naive_BM", "Sham_BM"),
  "NAIVE_BM_vs_UT_WB"        = c("Naive_BM", "UT_WB"),
  "NAIVE_BM_vs_UT_BM"        = c("Naive_BM", "UT_BM"),
  "SHAM_WB_vs_SHAM_BM"       = c("Sham_WB", "Sham_BM"),
  "SHAM_WB_vs_UT_WB"         = c("Sham_WB", "UT_WB"),
  "SHAM_WB_vs_UT_BM"         = c("Sham_WB", "UT_BM"),
  "SHAM_BM_vs_UT_WB"         = c("Sham_BM", "UT_WB"),
  "SHAM_BM_vs_UT_BM"         = c("Sham_BM", "UT_BM"),
  "UT_WB_vs_UT_BM"           = c("UT_WB", "UT_BM")
)

wb <- createWorkbook()

for (i in names(pairwise_comparisons)) {
  
  deg_T <- FindMarkers(seu_T, group.by = "sample_name", ident.1 = pairwise_comparisons[[i]][1], ident.2 = pairwise_comparisons[[i]][2], test.use = "MAST")
  
  addWorksheet(wb, sheetName = i)
  
  deg_T <- deg_T %>%
    filter(rownames(.) %in% selected_genes) %>%   # <- Filter only your selected genes
    arrange(-avg_log2FC)
  
  # Add gene names as a column
  deg_T$gene <- rownames(deg_T)
  deg_T <- deg_T %>% dplyr::select(gene, dplyr::everything())
  
  writeData(wb, sheet = i, x = deg_T)
}

saveWorkbook(wb, here("R Projects", "2107claude", "output", "de_analysis", "Find Markers", "DE_MAST_filtered.xlsx"), overwrite = TRUE)


```

#### Student's T Test

```{r}
# Define the gene list you want to focus on
selected_genes <- c("S1pr1", "S1pr2", "S1pr3", "S1pr4", "S1pr5", 
                    "Ccr7", "Sell", "Cd69", "Klf2", "Cxcr4", "Cd44", "Itgae", "Itgal")

pairwise_comparisons <- list(
  "NAIVE_WB_vs_NAIVE_BM"     = c("Naive_WB", "Naive_BM"),
  "NAIVE_WB_vs_SHAM_WB"      = c("Naive_WB", "Sham_WB"),
  "NAIVE_WB_vs_SHAM_BM"      = c("Naive_WB", "Sham_BM"),
  "NAIVE_WB_vs_UT_WB"        = c("Naive_WB", "UT_WB"),
  "NAIVE_WB_vs_UT_BM"        = c("Naive_WB", "UT_BM"),
  "NAIVE_BM_vs_SHAM_WB"      = c("Naive_BM", "Sham_WB"),
  "NAIVE_BM_vs_SHAM_BM"      = c("Naive_BM", "Sham_BM"),
  "NAIVE_BM_vs_UT_WB"        = c("Naive_BM", "UT_WB"),
  "NAIVE_BM_vs_UT_BM"        = c("Naive_BM", "UT_BM"),
  "SHAM_WB_vs_SHAM_BM"       = c("Sham_WB", "Sham_BM"),
  "SHAM_WB_vs_UT_WB"         = c("Sham_WB", "UT_WB"),
  "SHAM_WB_vs_UT_BM"         = c("Sham_WB", "UT_BM"),
  "SHAM_BM_vs_UT_WB"         = c("Sham_BM", "UT_WB"),
  "SHAM_BM_vs_UT_BM"         = c("Sham_BM", "UT_BM"),
  "UT_WB_vs_UT_BM"           = c("UT_WB", "UT_BM")
)

# Create an Excel workbook
wb <- createWorkbook()

for (i in names(pairwise_comparisons)) {
  
  deg_T <- FindMarkers(
    seu_T,
    group.by = "sample_name",
    ident.1 = pairwise_comparisons[[i]][1],
    ident.2 = pairwise_comparisons[[i]][2],
    test.use = "t"
  )
  
  addWorksheet(wb, sheetName = i)
  
  # Filter for selected genes and sort by log2FC
  deg_T <- deg_T %>%
    filter(rownames(.) %in% selected_genes) %>%
    arrange(-avg_log2FC)
  
  # Add gene names as a column and reorder
  deg_T$gene <- rownames(deg_T)
  deg_T <- deg_T %>% dplyr::select(gene, dplyr::everything())
  
  # Write to Excel sheet
  writeData(wb, sheet = i, x = deg_T)
}

# Save the workbook
saveWorkbook(
  wb,
  here("R Projects", "2107claude", "output", "de_analysis", "Find Markers", "DE_t_filtered.xlsx"),
  overwrite = TRUE
)
```

### Welch Two Sample t-test

#### S1pr1

```{r}
# Define the gene of interest
gene_of_interest <- "S1pr1"

# Define pairwise comparisons
pairwise_comparisons <- list(
  "NAIVE_WB_vs_NAIVE_BM"     = c("Naive_WB", "Naive_BM"),
  "NAIVE_WB_vs_SHAM_WB"      = c("Naive_WB", "Sham_WB"),
  "NAIVE_WB_vs_SHAM_BM"      = c("Naive_WB", "Sham_BM"),
  "NAIVE_WB_vs_UT_WB"        = c("Naive_WB", "UT_WB"),
  "NAIVE_WB_vs_UT_BM"        = c("Naive_WB", "UT_BM"),
  "NAIVE_BM_vs_SHAM_WB"      = c("Naive_BM", "Sham_WB"),
  "NAIVE_BM_vs_SHAM_BM"      = c("Naive_BM", "Sham_BM"),
  "NAIVE_BM_vs_UT_WB"        = c("Naive_BM", "UT_WB"),
  "NAIVE_BM_vs_UT_BM"        = c("Naive_BM", "UT_BM"),
  "SHAM_WB_vs_SHAM_BM"       = c("Sham_WB", "Sham_BM"),
  "SHAM_WB_vs_UT_WB"         = c("Sham_WB", "UT_WB"),
  "SHAM_WB_vs_UT_BM"         = c("Sham_WB", "UT_BM"),
  "SHAM_BM_vs_UT_WB"         = c("Sham_BM", "UT_WB"),
  "SHAM_BM_vs_UT_BM"         = c("Sham_BM", "UT_BM"),
  "UT_WB_vs_UT_BM"           = c("UT_WB", "UT_BM")
)

# Initialize list to store results for FDR calculation
all_pvalues <- numeric(length(pairwise_comparisons))
results_list <- list()

# Create workbook
wb <- createWorkbook()

# Loop through each pairwise comparison
for (i in seq_along(pairwise_comparisons)) {
  comparison_name <- names(pairwise_comparisons)[i]
  groups <- pairwise_comparisons[[i]]
  group1 <- groups[1]
  group2 <- groups[2]
  
  cat("Processing comparison:", comparison_name, "\n")
  
  # Extract expression data for the gene of interest
  gene_expression <- FetchData(seu_T, vars = gene_of_interest, layer = "data")
  
  # Get sample names from metadata
  sample_names <- seu_T@meta.data$sample_name
  
  # Extract expression values for each group
  group1_cells <- which(sample_names == group1)
  group2_cells <- which(sample_names == group2)
  
  group1_expression <- gene_expression[group1_cells, gene_of_interest]
  group2_expression <- gene_expression[group2_cells, gene_of_interest]
  
  # Check if both groups have data
  if (length(group1_expression) == 0 || length(group2_expression) == 0) {
    cat("Warning: No cells found for one or both groups in", comparison_name, "\n")
    next
  }
  
  # Perform Welch's t-test
  t_test_result <- t.test(group1_expression, group2_expression, var.equal = FALSE)
  
  # Calculate means
  mean_group1 <- mean(group1_expression, na.rm = TRUE)
  mean_group2 <- mean(group2_expression, na.rm = TRUE)
  
  # Calculate log2 fold change (group1 - group2)
  log2_fc <- mean_group1 - mean_group2
  
  # Extract results
  t_statistic <- t_test_result$statistic
  df <- t_test_result$parameter
  p_value <- t_test_result$p.value
  conf_int_lower <- t_test_result$conf.int[1]
  conf_int_upper <- t_test_result$conf.int[2]
  
  # Store p-value for FDR calculation
  all_pvalues[i] <- p_value
  
  # Create results data frame
  results_df <- data.frame(
    Gene = gene_of_interest,
    Comparison = comparison_name,
    Group1 = group1,
    Group2 = group2,
    N_Group1 = length(group1_expression),
    N_Group2 = length(group2_expression),
    Mean_Group1 = mean_group1,
    Mean_Group2 = mean_group2,
    Log2_Fold_Change = log2_fc,
    t_Statistic = as.numeric(t_statistic),
    Degrees_of_Freedom = as.numeric(df),
    P_Value = p_value,
    Conf_Int_Lower = conf_int_lower,
    Conf_Int_Upper = conf_int_upper,
    stringsAsFactors = FALSE
  )
  
  # Store results for later FDR adjustment
  results_list[[i]] <- results_df
  
  # Add worksheet to workbook
  addWorksheet(wb, sheetName = comparison_name)
  writeData(wb, sheet = comparison_name, x = results_df)
}

# Calculate FDR-adjusted p-values
fdr_adjusted <- p.adjust(all_pvalues, method = "fdr")

# Add FDR-adjusted p-values to each worksheet
for (i in seq_along(results_list)) {
  if (!is.null(results_list[[i]])) {
    comparison_name <- names(pairwise_comparisons)[i]
    
    # Add FDR column to existing data
    results_list[[i]]$FDR_Adjusted_P_Value <- fdr_adjusted[i]
    
    # Clear and rewrite the worksheet with updated data
    removeWorksheet(wb, comparison_name)
    addWorksheet(wb, sheetName = comparison_name)
    writeData(wb, sheet = comparison_name, x = results_list[[i]])
    
    # Print summary for this comparison
    cat("\n", comparison_name, ":\n")
    cat("  t =", round(results_list[[i]]$t_Statistic, 4), "\n")
    cat("  df =", round(results_list[[i]]$Degrees_of_Freedom, 2), "\n")
    cat("  p-value =", format(results_list[[i]]$P_Value, scientific = TRUE), "\n")
    cat("  FDR p-value =", format(results_list[[i]]$FDR_Adjusted_P_Value, scientific = TRUE), "\n")
    cat("  95% CI: [", round(results_list[[i]]$Conf_Int_Lower, 4), ", ", 
        round(results_list[[i]]$Conf_Int_Upper, 4), "]\n")
    cat("  Log2 FC =", round(results_list[[i]]$Log2_Fold_Change, 4), "\n")
  }
}

# Save the workbook
output_path <- here("R Projects", "2107claude", "output", "de_analysis", "t.test", paste0(gene_of_interest, ".xlsx"))

# Create directory if it doesn't exist
dir.create(dirname(output_path), recursive = TRUE, showWarnings = FALSE)

# Save workbook
saveWorkbook(wb, file = output_path, overwrite = TRUE)
```

#### S1pr2

```{r}
# Define the gene of interest
gene_of_interest <- "S1pr2"

# Define pairwise comparisons
pairwise_comparisons <- list(
  "NAIVE_WB_vs_NAIVE_BM"     = c("Naive_WB", "Naive_BM"),
  "NAIVE_WB_vs_SHAM_WB"      = c("Naive_WB", "Sham_WB"),
  "NAIVE_WB_vs_SHAM_BM"      = c("Naive_WB", "Sham_BM"),
  "NAIVE_WB_vs_UT_WB"        = c("Naive_WB", "UT_WB"),
  "NAIVE_WB_vs_UT_BM"        = c("Naive_WB", "UT_BM"),
  "NAIVE_BM_vs_SHAM_WB"      = c("Naive_BM", "Sham_WB"),
  "NAIVE_BM_vs_SHAM_BM"      = c("Naive_BM", "Sham_BM"),
  "NAIVE_BM_vs_UT_WB"        = c("Naive_BM", "UT_WB"),
  "NAIVE_BM_vs_UT_BM"        = c("Naive_BM", "UT_BM"),
  "SHAM_WB_vs_SHAM_BM"       = c("Sham_WB", "Sham_BM"),
  "SHAM_WB_vs_UT_WB"         = c("Sham_WB", "UT_WB"),
  "SHAM_WB_vs_UT_BM"         = c("Sham_WB", "UT_BM"),
  "SHAM_BM_vs_UT_WB"         = c("Sham_BM", "UT_WB"),
  "SHAM_BM_vs_UT_BM"         = c("Sham_BM", "UT_BM"),
  "UT_WB_vs_UT_BM"           = c("UT_WB", "UT_BM")
)

# Initialize list to store results for FDR calculation
all_pvalues <- numeric(length(pairwise_comparisons))
results_list <- list()

# Create workbook
wb <- createWorkbook()

# Loop through each pairwise comparison
for (i in seq_along(pairwise_comparisons)) {
  comparison_name <- names(pairwise_comparisons)[i]
  groups <- pairwise_comparisons[[i]]
  group1 <- groups[1]
  group2 <- groups[2]
  
  cat("Processing comparison:", comparison_name, "\n")
  
  # Extract expression data for the gene of interest
  gene_expression <- FetchData(seu_T, vars = gene_of_interest, layer = "data")
  
  # Get sample names from metadata
  sample_names <- seu_T@meta.data$sample_name
  
  # Extract expression values for each group
  group1_cells <- which(sample_names == group1)
  group2_cells <- which(sample_names == group2)
  
  group1_expression <- gene_expression[group1_cells, gene_of_interest]
  group2_expression <- gene_expression[group2_cells, gene_of_interest]
  
  # Check if both groups have data
  if (length(group1_expression) == 0 || length(group2_expression) == 0) {
    cat("Warning: No cells found for one or both groups in", comparison_name, "\n")
    next
  }
  
  # Perform Welch's t-test
  t_test_result <- t.test(group1_expression, group2_expression, var.equal = FALSE)
  
  # Calculate means
  mean_group1 <- mean(group1_expression, na.rm = TRUE)
  mean_group2 <- mean(group2_expression, na.rm = TRUE)
  
  # Calculate log2 fold change (group1 - group2)
  log2_fc <- mean_group1 - mean_group2
  
  # Extract results
  t_statistic <- t_test_result$statistic
  df <- t_test_result$parameter
  p_value <- t_test_result$p.value
  conf_int_lower <- t_test_result$conf.int[1]
  conf_int_upper <- t_test_result$conf.int[2]
  
  # Store p-value for FDR calculation
  all_pvalues[i] <- p_value
  
  # Create results data frame
  results_df <- data.frame(
    Gene = gene_of_interest,
    Comparison = comparison_name,
    Group1 = group1,
    Group2 = group2,
    N_Group1 = length(group1_expression),
    N_Group2 = length(group2_expression),
    Mean_Group1 = mean_group1,
    Mean_Group2 = mean_group2,
    Log2_Fold_Change = log2_fc,
    t_Statistic = as.numeric(t_statistic),
    Degrees_of_Freedom = as.numeric(df),
    P_Value = p_value,
    Conf_Int_Lower = conf_int_lower,
    Conf_Int_Upper = conf_int_upper,
    stringsAsFactors = FALSE
  )
  
  # Store results for later FDR adjustment
  results_list[[i]] <- results_df
  
  # Add worksheet to workbook
  addWorksheet(wb, sheetName = comparison_name)
  writeData(wb, sheet = comparison_name, x = results_df)
}

# Calculate FDR-adjusted p-values
fdr_adjusted <- p.adjust(all_pvalues, method = "fdr")

# Add FDR-adjusted p-values to each worksheet
for (i in seq_along(results_list)) {
  if (!is.null(results_list[[i]])) {
    comparison_name <- names(pairwise_comparisons)[i]
    
    # Add FDR column to existing data
    results_list[[i]]$FDR_Adjusted_P_Value <- fdr_adjusted[i]
    
    # Clear and rewrite the worksheet with updated data
    removeWorksheet(wb, comparison_name)
    addWorksheet(wb, sheetName = comparison_name)
    writeData(wb, sheet = comparison_name, x = results_list[[i]])
    
    # Print summary for this comparison
    cat("\n", comparison_name, ":\n")
    cat("  t =", round(results_list[[i]]$t_Statistic, 4), "\n")
    cat("  df =", round(results_list[[i]]$Degrees_of_Freedom, 2), "\n")
    cat("  p-value =", format(results_list[[i]]$P_Value, scientific = TRUE), "\n")
    cat("  FDR p-value =", format(results_list[[i]]$FDR_Adjusted_P_Value, scientific = TRUE), "\n")
    cat("  95% CI: [", round(results_list[[i]]$Conf_Int_Lower, 4), ", ", 
        round(results_list[[i]]$Conf_Int_Upper, 4), "]\n")
    cat("  Log2 FC =", round(results_list[[i]]$Log2_Fold_Change, 4), "\n")
  }
}

# Save the workbook
output_path <- here("R Projects", "2107claude", "output", "de_analysis", "t.test", paste0(gene_of_interest, ".xlsx"))

# Create directory if it doesn't exist
dir.create(dirname(output_path), recursive = TRUE, showWarnings = FALSE)

# Save workbook
saveWorkbook(wb, file = output_path, overwrite = TRUE)
```

#### S1pr3

```{r}
# Define the gene of interest
gene_of_interest <- "S1pr3"

# Define pairwise comparisons
pairwise_comparisons <- list(
  "NAIVE_WB_vs_NAIVE_BM"     = c("Naive_WB", "Naive_BM"),
  "NAIVE_WB_vs_SHAM_WB"      = c("Naive_WB", "Sham_WB"),
  "NAIVE_WB_vs_SHAM_BM"      = c("Naive_WB", "Sham_BM"),
  "NAIVE_WB_vs_UT_WB"        = c("Naive_WB", "UT_WB"),
  "NAIVE_WB_vs_UT_BM"        = c("Naive_WB", "UT_BM"),
  "NAIVE_BM_vs_SHAM_WB"      = c("Naive_BM", "Sham_WB"),
  "NAIVE_BM_vs_SHAM_BM"      = c("Naive_BM", "Sham_BM"),
  "NAIVE_BM_vs_UT_WB"        = c("Naive_BM", "UT_WB"),
  "NAIVE_BM_vs_UT_BM"        = c("Naive_BM", "UT_BM"),
  "SHAM_WB_vs_SHAM_BM"       = c("Sham_WB", "Sham_BM"),
  "SHAM_WB_vs_UT_WB"         = c("Sham_WB", "UT_WB"),
  "SHAM_WB_vs_UT_BM"         = c("Sham_WB", "UT_BM"),
  "SHAM_BM_vs_UT_WB"         = c("Sham_BM", "UT_WB"),
  "SHAM_BM_vs_UT_BM"         = c("Sham_BM", "UT_BM"),
  "UT_WB_vs_UT_BM"           = c("UT_WB", "UT_BM")
)

# Initialize list to store results for FDR calculation
all_pvalues <- numeric(length(pairwise_comparisons))
results_list <- list()

# Create workbook
wb <- createWorkbook()

# Loop through each pairwise comparison
for (i in seq_along(pairwise_comparisons)) {
  comparison_name <- names(pairwise_comparisons)[i]
  groups <- pairwise_comparisons[[i]]
  group1 <- groups[1]
  group2 <- groups[2]
  
  cat("Processing comparison:", comparison_name, "\n")
  
  # Extract expression data for the gene of interest
  gene_expression <- FetchData(seu_T, vars = gene_of_interest, layer = "data")
  
  # Get sample names from metadata
  sample_names <- seu_T@meta.data$sample_name
  
  # Extract expression values for each group
  group1_cells <- which(sample_names == group1)
  group2_cells <- which(sample_names == group2)
  
  group1_expression <- gene_expression[group1_cells, gene_of_interest]
  group2_expression <- gene_expression[group2_cells, gene_of_interest]
  
  # Check if both groups have data
  if (length(group1_expression) == 0 || length(group2_expression) == 0) {
    cat("Warning: No cells found for one or both groups in", comparison_name, "\n")
    next
  }
  
  # Perform Welch's t-test
  t_test_result <- t.test(group1_expression, group2_expression, var.equal = FALSE)
  
  # Calculate means
  mean_group1 <- mean(group1_expression, na.rm = TRUE)
  mean_group2 <- mean(group2_expression, na.rm = TRUE)
  
  # Calculate log2 fold change (group1 - group2)
  log2_fc <- mean_group1 - mean_group2
  
  # Extract results
  t_statistic <- t_test_result$statistic
  df <- t_test_result$parameter
  p_value <- t_test_result$p.value
  conf_int_lower <- t_test_result$conf.int[1]
  conf_int_upper <- t_test_result$conf.int[2]
  
  # Store p-value for FDR calculation
  all_pvalues[i] <- p_value
  
  # Create results data frame
  results_df <- data.frame(
    Gene = gene_of_interest,
    Comparison = comparison_name,
    Group1 = group1,
    Group2 = group2,
    N_Group1 = length(group1_expression),
    N_Group2 = length(group2_expression),
    Mean_Group1 = mean_group1,
    Mean_Group2 = mean_group2,
    Log2_Fold_Change = log2_fc,
    t_Statistic = as.numeric(t_statistic),
    Degrees_of_Freedom = as.numeric(df),
    P_Value = p_value,
    Conf_Int_Lower = conf_int_lower,
    Conf_Int_Upper = conf_int_upper,
    stringsAsFactors = FALSE
  )
  
  # Store results for later FDR adjustment
  results_list[[i]] <- results_df
  
  # Add worksheet to workbook
  addWorksheet(wb, sheetName = comparison_name)
  writeData(wb, sheet = comparison_name, x = results_df)
}

# Calculate FDR-adjusted p-values
fdr_adjusted <- p.adjust(all_pvalues, method = "fdr")

# Add FDR-adjusted p-values to each worksheet
for (i in seq_along(results_list)) {
  if (!is.null(results_list[[i]])) {
    comparison_name <- names(pairwise_comparisons)[i]
    
    # Add FDR column to existing data
    results_list[[i]]$FDR_Adjusted_P_Value <- fdr_adjusted[i]
    
    # Clear and rewrite the worksheet with updated data
    removeWorksheet(wb, comparison_name)
    addWorksheet(wb, sheetName = comparison_name)
    writeData(wb, sheet = comparison_name, x = results_list[[i]])
    
    # Print summary for this comparison
    cat("\n", comparison_name, ":\n")
    cat("  t =", round(results_list[[i]]$t_Statistic, 4), "\n")
    cat("  df =", round(results_list[[i]]$Degrees_of_Freedom, 2), "\n")
    cat("  p-value =", format(results_list[[i]]$P_Value, scientific = TRUE), "\n")
    cat("  FDR p-value =", format(results_list[[i]]$FDR_Adjusted_P_Value, scientific = TRUE), "\n")
    cat("  95% CI: [", round(results_list[[i]]$Conf_Int_Lower, 4), ", ", 
        round(results_list[[i]]$Conf_Int_Upper, 4), "]\n")
    cat("  Log2 FC =", round(results_list[[i]]$Log2_Fold_Change, 4), "\n")
  }
}

# Save the workbook
output_path <- here("R Projects", "2107claude", "output", "de_analysis", "t.test", paste0(gene_of_interest, ".xlsx"))

# Create directory if it doesn't exist
dir.create(dirname(output_path), recursive = TRUE, showWarnings = FALSE)

# Save workbook
saveWorkbook(wb, file = output_path, overwrite = TRUE)
```

#### S1pr4

```{r}
# Define the gene of interest
gene_of_interest <- "S1pr4"

# Define pairwise comparisons
pairwise_comparisons <- list(
  "NAIVE_WB_vs_NAIVE_BM"     = c("Naive_WB", "Naive_BM"),
  "NAIVE_WB_vs_SHAM_WB"      = c("Naive_WB", "Sham_WB"),
  "NAIVE_WB_vs_SHAM_BM"      = c("Naive_WB", "Sham_BM"),
  "NAIVE_WB_vs_UT_WB"        = c("Naive_WB", "UT_WB"),
  "NAIVE_WB_vs_UT_BM"        = c("Naive_WB", "UT_BM"),
  "NAIVE_BM_vs_SHAM_WB"      = c("Naive_BM", "Sham_WB"),
  "NAIVE_BM_vs_SHAM_BM"      = c("Naive_BM", "Sham_BM"),
  "NAIVE_BM_vs_UT_WB"        = c("Naive_BM", "UT_WB"),
  "NAIVE_BM_vs_UT_BM"        = c("Naive_BM", "UT_BM"),
  "SHAM_WB_vs_SHAM_BM"       = c("Sham_WB", "Sham_BM"),
  "SHAM_WB_vs_UT_WB"         = c("Sham_WB", "UT_WB"),
  "SHAM_WB_vs_UT_BM"         = c("Sham_WB", "UT_BM"),
  "SHAM_BM_vs_UT_WB"         = c("Sham_BM", "UT_WB"),
  "SHAM_BM_vs_UT_BM"         = c("Sham_BM", "UT_BM"),
  "UT_WB_vs_UT_BM"           = c("UT_WB", "UT_BM")
)

# Initialize list to store results for FDR calculation
all_pvalues <- numeric(length(pairwise_comparisons))
results_list <- list()

# Create workbook
wb <- createWorkbook()

# Loop through each pairwise comparison
for (i in seq_along(pairwise_comparisons)) {
  comparison_name <- names(pairwise_comparisons)[i]
  groups <- pairwise_comparisons[[i]]
  group1 <- groups[1]
  group2 <- groups[2]
  
  cat("Processing comparison:", comparison_name, "\n")
  
  # Extract expression data for the gene of interest
  gene_expression <- FetchData(seu_T, vars = gene_of_interest, layer = "data")
  
  # Get sample names from metadata
  sample_names <- seu_T@meta.data$sample_name
  
  # Extract expression values for each group
  group1_cells <- which(sample_names == group1)
  group2_cells <- which(sample_names == group2)
  
  group1_expression <- gene_expression[group1_cells, gene_of_interest]
  group2_expression <- gene_expression[group2_cells, gene_of_interest]
  
  # Check if both groups have data
  if (length(group1_expression) == 0 || length(group2_expression) == 0) {
    cat("Warning: No cells found for one or both groups in", comparison_name, "\n")
    next
  }
  
  # Perform Welch's t-test
  t_test_result <- t.test(group1_expression, group2_expression, var.equal = FALSE)
  
  # Calculate means
  mean_group1 <- mean(group1_expression, na.rm = TRUE)
  mean_group2 <- mean(group2_expression, na.rm = TRUE)
  
  # Calculate log2 fold change (group1 - group2)
  log2_fc <- mean_group1 - mean_group2
  
  # Extract results
  t_statistic <- t_test_result$statistic
  df <- t_test_result$parameter
  p_value <- t_test_result$p.value
  conf_int_lower <- t_test_result$conf.int[1]
  conf_int_upper <- t_test_result$conf.int[2]
  
  # Store p-value for FDR calculation
  all_pvalues[i] <- p_value
  
  # Create results data frame
  results_df <- data.frame(
    Gene = gene_of_interest,
    Comparison = comparison_name,
    Group1 = group1,
    Group2 = group2,
    N_Group1 = length(group1_expression),
    N_Group2 = length(group2_expression),
    Mean_Group1 = mean_group1,
    Mean_Group2 = mean_group2,
    Log2_Fold_Change = log2_fc,
    t_Statistic = as.numeric(t_statistic),
    Degrees_of_Freedom = as.numeric(df),
    P_Value = p_value,
    Conf_Int_Lower = conf_int_lower,
    Conf_Int_Upper = conf_int_upper,
    stringsAsFactors = FALSE
  )
  
  # Store results for later FDR adjustment
  results_list[[i]] <- results_df
  
  # Add worksheet to workbook
  addWorksheet(wb, sheetName = comparison_name)
  writeData(wb, sheet = comparison_name, x = results_df)
}

# Calculate FDR-adjusted p-values
fdr_adjusted <- p.adjust(all_pvalues, method = "fdr")

# Add FDR-adjusted p-values to each worksheet
for (i in seq_along(results_list)) {
  if (!is.null(results_list[[i]])) {
    comparison_name <- names(pairwise_comparisons)[i]
    
    # Add FDR column to existing data
    results_list[[i]]$FDR_Adjusted_P_Value <- fdr_adjusted[i]
    
    # Clear and rewrite the worksheet with updated data
    removeWorksheet(wb, comparison_name)
    addWorksheet(wb, sheetName = comparison_name)
    writeData(wb, sheet = comparison_name, x = results_list[[i]])
    
    # Print summary for this comparison
    cat("\n", comparison_name, ":\n")
    cat("  t =", round(results_list[[i]]$t_Statistic, 4), "\n")
    cat("  df =", round(results_list[[i]]$Degrees_of_Freedom, 2), "\n")
    cat("  p-value =", format(results_list[[i]]$P_Value, scientific = TRUE), "\n")
    cat("  FDR p-value =", format(results_list[[i]]$FDR_Adjusted_P_Value, scientific = TRUE), "\n")
    cat("  95% CI: [", round(results_list[[i]]$Conf_Int_Lower, 4), ", ", 
        round(results_list[[i]]$Conf_Int_Upper, 4), "]\n")
    cat("  Log2 FC =", round(results_list[[i]]$Log2_Fold_Change, 4), "\n")
  }
}

# Save the workbook
output_path <- here("R Projects", "2107", "output", "de_analysis", "t.test", paste0(gene_of_interest, ".xlsx"))

# Create directory if it doesn't exist
dir.create(dirname(output_path), recursive = TRUE, showWarnings = FALSE)

# Save workbook
saveWorkbook(wb, file = output_path, overwrite = TRUE)
```

# Visualisation

## T Cells

### Volcano Plots

```{R, cahce=TRUE}
# Function to create volcano plots for S1PR genes
create_s1pr_volcano_plots <- function(seu_T) {
  # Define S1PR genes
  s1pr_genes <- c("S1pr1", "S1pr2", "S1pr3", "S1pr4", "S1pr5")
  
  # Define all 15 pairwise comparisons
  pairwise_comparisons <- list(
    "NAIVE_WB_vs_NAIVE_BM"     = c("Naive_WB", "Naive_BM"),
    "NAIVE_WB_vs_SHAM_WB"      = c("Naive_WB", "Sham_WB"),
    "NAIVE_WB_vs_SHAM_BM"      = c("Naive_WB", "Sham_BM"),
    "NAIVE_WB_vs_UT_WB"        = c("Naive_WB", "UT_WB"),
    "NAIVE_WB_vs_UT_BM"        = c("Naive_WB", "UT_BM"),
    "NAIVE_BM_vs_SHAM_WB"      = c("Naive_BM", "Sham_WB"),
    "NAIVE_BM_vs_SHAM_BM"      = c("Naive_BM", "Sham_BM"),
    "NAIVE_BM_vs_UT_WB"        = c("Naive_BM", "UT_WB"),
    "NAIVE_BM_vs_UT_BM"        = c("Naive_BM", "UT_BM"),
    "SHAM_WB_vs_SHAM_BM"       = c("Sham_WB", "Sham_BM"),
    "SHAM_WB_vs_UT_WB"         = c("Sham_WB", "UT_WB"),
    "SHAM_WB_vs_UT_BM"         = c("Sham_WB", "UT_BM"),
    "SHAM_BM_vs_UT_WB"         = c("Sham_BM", "UT_WB"),
    "SHAM_BM_vs_UT_BM"         = c("Sham_BM", "UT_BM"),
    "UT_WB_vs_UT_BM"           = c("UT_WB", "UT_BM")
  )
  
  volcano_plots <- list()
  
  for (comparison_name in names(pairwise_comparisons)) {
    groups <- pairwise_comparisons[[comparison_name]]
    
    cat("Creating volcano plot for:", comparison_name, "\n")
    
    # Run differential expression analysis
    de_results <- FindMarkers(
      seu_T,
      group.by = "sample_name",
      ident.1 = groups[1],
      ident.2 = groups[2],
      test.use = "wilcox",
      logfc.threshold = 0,
      min.pct = 0
    )
    
    # Add gene names
    de_results$gene <- rownames(de_results)
    
    # Fix zero p-values to avoid volcano plot warnings
    # Replace 0 p-values with a very small number
    min_nonzero_p <- min(de_results$p_val[de_results$p_val > 0], na.rm = TRUE)
    min_nonzero_p_adj <- min(de_results$p_val_adj[de_results$p_val_adj > 0], na.rm = TRUE)
    
    de_results$p_val[de_results$p_val == 0] <- min_nonzero_p * 0.1
    de_results$p_val_adj[de_results$p_val_adj == 0] <- min_nonzero_p_adj * 0.1
    
    # Create volcano plot
    volcano_plot <- EnhancedVolcano(
      de_results,
      lab = de_results$gene,
      x = 'avg_log2FC',
      y = 'p_val_adj',
      title = paste('Volcano Plot:', comparison_name),
      subtitle = paste('T Cells:', groups[1], 'vs', groups[2]),
      pCutoff = 0.05,
      FCcutoff = 0.25,
      pointSize = 1.5,
      labSize = 3,
      # Highlight S1PR genes
      selectLab = s1pr_genes,
      boxedLabels = TRUE,
      colAlpha = 0.7,
      legendPosition = 'right',
      legendLabSize = 10,
      legendIconSize = 3.0,
      drawConnectors = TRUE,
      widthConnectors = 0.3,
      colConnectors = 'black'
    )
    
    volcano_plots[[comparison_name]] <- volcano_plot
    print(volcano_plot)
    
    # Save individual volcano plot as SVG
    ggsave(
      filename = here("R Projects", "2107claude", "output", "visualisation", "volcano", 
                      paste0("volcano_", comparison_name, ".svg")),
      plot = volcano_plot,
      device = "svg",
      width = 10, height = 8
    )
  }
  
  cat("\nTotal volcano plots created:", length(volcano_plots), "\n")
  
  return(volcano_plots)
}

# Create volcano plots
volcano_plots <- create_s1pr_volcano_plots(seu_T)
```

### Box Plots

```{R, cache=TRUE}
# Function to create individual box plots for each S1PR gene
create_s1pr_box_plots <- function(seu_T) {
  # Define S1PR genes
  s1pr_genes <- c("S1pr1", "S1pr2", "S1pr3", "S1pr4", "S1pr5")
  
  # Check which genes are available
  available_genes <- s1pr_genes[s1pr_genes %in% rownames(seu_T)]
  
  if (length(available_genes) == 0) {
    cat("No S1PR genes found in dataset\n")
    return(NULL)
  }
  
  cat("Creating individual box plots for genes:", paste(available_genes, collapse = ", "), "\n")
  
  # Extract expression data
  expr_data <- GetAssayData(seu_T, slot = "data", layer = "data")[available_genes, , drop = FALSE]
  if (inherits(expr_data, "dgCMatrix")) {
    expr_data <- as.matrix(expr_data)
  }
  
  # Create long format data
  plot_data <- expr_data %>%
    as.data.frame() %>%
    tibble::rownames_to_column("gene") %>%
    pivot_longer(-gene, names_to = "cell", values_to = "expression") %>%
    left_join(seu_T@meta.data %>% tibble::rownames_to_column("cell"), by = "cell")
  
  # Define colors for your actual data values
  treatment_colors <- c(
    "NAIVE" = "#2166AC",
    "SHAM" = "#762A83", 
    "UT" = "#C51B7D"
  )
  
  tissue_colors <- c(
    "Whole_blood" = "#D73027",
    "Bone_marrow" = "#4575B4"
  )
  
  all_plots <- list()
  
  # Create individual plots for each gene
  for (gene in available_genes) {
    gene_data <- plot_data %>% filter(gene == !!gene)
    
    cat("Creating box plots for", gene, "\n")
    
    # Box plot by treatment for this gene
    p_treatment <- ggplot(gene_data, aes(x = treatment, y = expression, fill = treatment)) +
      geom_boxplot(alpha = 0.7, outlier.size = 0.5) +
      geom_jitter(width = 0.2, alpha = 0.2, size = 0.3) +
      scale_fill_manual(values = treatment_colors) +
      theme_minimal() +
      theme(
        axis.text.x = element_text(angle = 45, hjust = 1, size = 12),
        axis.text.y = element_text(size = 11),
        axis.title = element_text(size = 14, face = "bold"),
        plot.title = element_text(size = 16, face = "bold"),
        plot.subtitle = element_text(size = 12),
        legend.position = "none",
        panel.grid.minor = element_blank()
      ) +
      labs(
        x = "Treatment",
        y = "Expression (Log-normalized)",
        title = paste(gene, "Expression by Treatment"),
        subtitle = "T Cells"
      )
    
    # Box plot by tissue for this gene
    p_tissue <- ggplot(gene_data, aes(x = tissue, y = expression, fill = tissue)) +
      geom_boxplot(alpha = 0.7, outlier.size = 0.5) +
      geom_jitter(width = 0.2, alpha = 0.2, size = 0.3) +
      scale_fill_manual(values = tissue_colors) +
      theme_minimal() +
      theme(
        axis.text.x = element_text(angle = 45, hjust = 1, size = 12),
        axis.text.y = element_text(size = 11),
        axis.title = element_text(size = 14, face = "bold"),
        plot.title = element_text(size = 16, face = "bold"),
        plot.subtitle = element_text(size = 12),
        legend.position = "none",
        panel.grid.minor = element_blank()
      ) +
      labs(
        x = "Tissue",
        y = "Expression (Log-normalized)",
        title = paste(gene, "Expression by Tissue"),
        subtitle = "T Cells"
      )
    
    # Box plot by sample for this gene
    p_sample <- ggplot(gene_data, aes(x = sample_name, y = expression, fill = treatment)) +
      geom_boxplot(alpha = 0.7, outlier.size = 0.5) +
      scale_fill_manual(values = treatment_colors) +
      theme_minimal() +
      theme(
        axis.text.x = element_text(angle = 45, hjust = 1, size = 11),
        axis.text.y = element_text(size = 11),
        axis.title = element_text(size = 14, face = "bold"),
        plot.title = element_text(size = 16, face = "bold"),
        plot.subtitle = element_text(size = 12),
        legend.title = element_text(size = 12, face = "bold"),
        legend.text = element_text(size = 10),
        panel.grid.minor = element_blank()
      ) +
      labs(
        x = "Sample",
        y = "Expression (Log-normalized)",
        title = paste(gene, "Expression by Sample"),
        subtitle = "T Cells",
        fill = "Treatment"
      )
    
    # Display plots
    print(p_treatment)
    print(p_tissue)
    print(p_sample)
    
    # Save individual plots as SVG
    ggsave(here("R Projects", "2107claude", "output", "visualisation", "box plots", 
                paste0(gene, "_boxplot_treatment.svg")),
           p_treatment, device = "svg", width = 8, height = 6)
    ggsave(here("R Projects", "2107claude", "output", "visualisation", "box plots", 
                paste0(gene, "_boxplot_tissue.svg")),
           p_tissue, device = "svg", width = 8, height = 6)
    ggsave(here("R Projects", "2107claude", "output", "visualisation", "box plots", 
                paste0(gene, "_boxplot_sample.svg")),
           p_sample, device = "svg", width = 10, height = 6)
    
    # Store plots
    all_plots[[paste0(gene, "_treatment")]] <- p_treatment
    all_plots[[paste0(gene, "_tissue")]] <- p_tissue
    all_plots[[paste0(gene, "_sample")]] <- p_sample
  }
  
  return(all_plots)
}

# Create box plots
box_plots <- create_s1pr_box_plots(seu_T)
```

### Violin Plots

```{R, cache=TRUE}
# Function to create individual violin plots for each S1PR gene
create_s1pr_violin_plots <- function(seu_T) {
  # Define S1PR genes
  s1pr_genes <- c("S1pr1", "S1pr2", "S1pr3", "S1pr4", "S1pr5")
  
  # Check which genes are available
  available_genes <- s1pr_genes[s1pr_genes %in% rownames(seu_T)]
  
  if (length(available_genes) == 0) {
    cat("No S1PR genes found in dataset\n")
    return(NULL)
  }
  
  cat("Creating individual violin plots for genes:", paste(available_genes, collapse = ", "), "\n")
  
  # Extract expression data
  expr_data <- GetAssayData(seu_T, slot = "data", layer = "data")[available_genes, , drop = FALSE]
  if (inherits(expr_data, "dgCMatrix")) {
    expr_data <- as.matrix(expr_data)
  }
  
  # Create long format data
  plot_data <- expr_data %>%
    as.data.frame() %>%
    tibble::rownames_to_column("gene") %>%
    pivot_longer(-gene, names_to = "cell", values_to = "expression") %>%
    left_join(seu_T@meta.data %>% tibble::rownames_to_column("cell"), by = "cell")
  
  # Define colors for your actual data values
  treatment_colors <- c(
    "NAIVE" = "#2166AC",
    "SHAM" = "#762A83", 
    "UT" = "#C51B7D"
  )
  
  tissue_colors <- c(
    "Whole_blood" = "#D73027",
    "Bone_marrow" = "#4575B4"
  )
  
  all_plots <- list()
  
  # Create individual plots for each gene
  for (gene in available_genes) {
    gene_data <- plot_data %>% filter(gene == !!gene)
    
    cat("Creating violin plots for", gene, "\n")
    
    # Violin plot by treatment for this gene
    p_violin_treatment <- ggplot(gene_data, aes(x = treatment, y = expression, fill = treatment)) +
      geom_violin(scale = "width", alpha = 0.7, trim = FALSE) +
      geom_boxplot(width = 0.1, outlier.shape = NA, alpha = 0.8, fill = "white") +
      stat_summary(fun = median, geom = "point", size = 3, color = "black") +
      scale_fill_manual(values = treatment_colors) +
      theme_minimal() +
      theme(
        axis.text.x = element_text(angle = 45, hjust = 1, size = 12),
        axis.text.y = element_text(size = 11),
        axis.title = element_text(size = 14, face = "bold"),
        plot.title = element_text(size = 16, face = "bold"),
        plot.subtitle = element_text(size = 12),
        legend.position = "none",
        panel.grid.minor = element_blank()
      ) +
      labs(
        x = "Treatment",
        y = "Expression (Log-normalized)",
        title = paste(gene, "Expression Distribution by Treatment"),
        subtitle = "T Cells - Violin Plot"
      )
    
    # Violin plot by sample for this gene
    p_violin_sample <- ggplot(gene_data, aes(x = sample_name, y = expression, fill = treatment)) +
      geom_violin(scale = "width", alpha = 0.7, trim = FALSE) +
      geom_boxplot(width = 0.1, outlier.shape = NA, alpha = 0.8, fill = "white") +
      scale_fill_manual(values = treatment_colors) +
      theme_minimal() +
      theme(
        axis.text.x = element_text(angle = 45, hjust = 1, size = 11),
        axis.text.y = element_text(size = 11),
        axis.title = element_text(size = 14, face = "bold"),
        plot.title = element_text(size = 16, face = "bold"),
        plot.subtitle = element_text(size = 12),
        legend.title = element_text(size = 12, face = "bold"),
        legend.text = element_text(size = 10),
        panel.grid.minor = element_blank()
      ) +
      labs(
        x = "Sample",
        y = "Expression (Log-normalized)",
        title = paste(gene, "Expression Distribution by Sample"),
        subtitle = "T Cells - Violin Plot",
        fill = "Treatment"
      )
    
    # Split violin plot comparing tissues within each treatment for this gene
    p_violin_split <- ggplot(gene_data, aes(x = treatment, y = expression, fill = tissue)) +
      geom_violin(position = position_dodge(width = 0.8), alpha = 0.7) +
      geom_boxplot(position = position_dodge(width = 0.8), width = 0.1, 
                   outlier.shape = NA, alpha = 0.8, fill = "white") +
      scale_fill_manual(values = tissue_colors) +
      theme_minimal() +
      theme(
        axis.text.x = element_text(angle = 45, hjust = 1, size = 12),
        axis.text.y = element_text(size = 11),
        axis.title = element_text(size = 14, face = "bold"),
        plot.title = element_text(size = 16, face = "bold"),
        plot.subtitle = element_text(size = 12),
        legend.title = element_text(size = 12, face = "bold"),
        legend.text = element_text(size = 10),
        panel.grid.minor = element_blank()
      ) +
      labs(
        x = "Treatment",
        y = "Expression (Log-normalized)",
        title = paste(gene, "Expression: Treatment vs Tissue"),
        subtitle = "T Cells - Split Violin Plot",
        fill = "Tissue"
      )
    
    # Display plots
    print(p_violin_treatment)
    print(p_violin_sample)
    print(p_violin_split)
    
    # Save individual plots as SVG
    ggsave(here("R Projects", "2107claude", "output", "visualisation", "violin plots", 
                paste0(gene, "_violin_treatment.svg")),
           p_violin_treatment, device = "svg", width = 8, height = 6)
    ggsave(here("R Projects", "2107claude", "output", "visualisation", "violin plots", 
                paste0(gene, "_violin_sample.svg")),
           p_violin_sample, device = "svg", width = 10, height = 6)
    ggsave(here("R Projects", "2107claude", "output", "visualisation", "violin plots", 
                paste0(gene, "_violin_split.svg")),
           p_violin_split, device = "svg", width = 8, height = 6)
    
    # Store plots
    all_plots[[paste0(gene, "_treatment")]] <- p_violin_treatment
    all_plots[[paste0(gene, "_sample")]] <- p_violin_sample
    all_plots[[paste0(gene, "_split")]] <- p_violin_split
  }
  
  return(all_plots)
}

# Create violin plots
violin_plots <- create_s1pr_violin_plots(seu_T)
```

### Nebulosa

```{R, cache=TRUE}
# S1pr1
Nebulosa::plot_density(seu_T, "S1pr1") + 
    facet_wrap(.~seu_T$sample_name, ncol = 3) + theme_void()
# S1pr2
Nebulosa::plot_density(seu_T, "S1pr2") + 
    facet_wrap(.~seu_T$sample_name, ncol = 3) + theme_void()
# S1pr3
Nebulosa::plot_density(seu_T, "S1pr3") + 
    facet_wrap(.~seu_T$sample_name, ncol = 3) + theme_void()
# S1pr4
Nebulosa::plot_density(seu_T, "S1pr4") + 
    facet_wrap(.~seu_T$sample_name, ncol = 3) + theme_void()
# S1pr5
Nebulosa::plot_density(seu_T, "S1pr5") + 
    facet_wrap(.~seu_T$sample_name, ncol = 3) + theme_void()
```

```{r}

```

### Heatmaps

```{R, cache=TRUE, cache.lazy=FALSE}
# Define S1PR gene family
s1pr_genes <- c("S1pr1", "S1pr2", "S1pr3", "S1pr4", "S1pr5")

# Function to check gene availability and get expression data
extract_s1pr_data <- function(seurat_obj, genes = s1pr_genes) {
  # Check which S1PR genes are present in the dataset
  available_genes <- genes[genes %in% rownames(seurat_obj)]
  missing_genes <- genes[!genes %in% rownames(seurat_obj)]
  
  if (length(missing_genes) > 0) {
    cat("Warning: Missing genes:", paste(missing_genes, collapse = ", "), "\n")
  }
  
  if (length(available_genes) == 0) {
    stop("No S1PR genes found in the dataset!")
  }
  
  cat("Found S1PR genes:", paste(available_genes, collapse = ", "), "\n")
  
  # Extract expression data (using layer parameter instead of deprecated slot)
  expr_data <- GetAssayData(seurat_obj, layer = "data")[available_genes, , drop = FALSE]
  
  # Convert sparse matrix to dense if needed
  if (inherits(expr_data, "dgCMatrix")) {
    expr_data <- as.matrix(expr_data)
  }
  
  # Get metadata
  metadata <- seurat_obj@meta.data
  
  return(list(
    expression = expr_data,
    metadata = metadata,
    available_genes = available_genes
  ))
}

# Function to create sample annotation colors
create_annotation_colors <- function(metadata) {
  # Define color palettes for your specific conditions
  treatment_colors <- c(
    "NAIVE" = "#2166AC",
    "SHAM" = "#762A83", 
    "UT" = "#C51B7D"
  )
  
  tissue_colors <- c(
    "Whole_blood" = "#D73027",
    "Bone_marrow" = "#4575B4"
  )
  
  return(list(
    treatment = treatment_colors,
    tissue = tissue_colors
  ))
}

# Function to calculate summary statistics by treatment and tissue
calculate_treatment_tissue_stats <- function(expr_data, metadata) {
  # Convert to long format for easier manipulation
  expr_long <- expr_data %>%
    as.data.frame() %>%
    tibble::rownames_to_column("gene") %>%
    tidyr::pivot_longer(-gene, names_to = "cell", values_to = "expression")
  
  # Add metadata
  expr_long <- expr_long %>%
    dplyr::left_join(metadata %>% tibble::rownames_to_column("cell"), by = "cell")
  
  # Calculate summary statistics by treatment, tissue, and gene
  summary_stats <- expr_long %>%
    dplyr::group_by(gene, treatment, tissue) %>%
    dplyr::summarise(
      mean_expr = mean(expression),
      median_expr = median(expression),
      sd_expr = sd(expression),
      n_cells = n(),
      pct_positive = sum(expression > 0) / n() * 100,
      .groups = "drop"
    ) %>%
    # Create combined treatment_tissue column for better visualization
    dplyr::mutate(treatment_tissue = paste(treatment, tissue, sep = "_"))
  
  return(summary_stats)
}

# Function to create S1PR expression heatmap by treatment and tissue
create_s1pr_treatment_tissue_heatmap <- function(seurat_obj, metric = "mean") {
  # Extract data
  s1pr_data <- extract_s1pr_data(seurat_obj)
  expr_data <- s1pr_data$expression
  metadata <- s1pr_data$metadata
  
  # Calculate summary statistics
  summary_data <- calculate_treatment_tissue_stats(expr_data, metadata)
  
  cat("Summary data structure:\n")
  print(head(summary_data))
  cat("Unique treatment_tissue combinations:", paste(unique(summary_data$treatment_tissue), collapse = ", "), "\n")
  
  # Create summary matrix using explicit dplyr functions
  if (metric == "mean") {
    summary_matrix <- summary_data %>%
      dplyr::select(gene, treatment_tissue, mean_expr) %>%
      tidyr::pivot_wider(names_from = treatment_tissue, values_from = mean_expr) %>%
      tibble::column_to_rownames("gene") %>%
      as.matrix()
    title_suffix <- "Mean Expression"
    legend_title <- "Mean Expression"
  } else if (metric == "pct_positive") {
    summary_matrix <- summary_data %>%
      dplyr::select(gene, treatment_tissue, pct_positive) %>%
      tidyr::pivot_wider(names_from = treatment_tissue, values_from = pct_positive) %>%
      tibble::column_to_rownames("gene") %>%
      as.matrix()
    title_suffix <- "% Positive Cells"
    legend_title <- "% Positive"
  }
  
  cat("Summary matrix dimensions:", dim(summary_matrix), "\n")
  cat("Column names:", colnames(summary_matrix), "\n")
  
  # Create annotation for treatment-tissue combinations
  col_info <- summary_data %>%
    dplyr::select(treatment_tissue, treatment, tissue) %>%
    dplyr::distinct() %>%
    dplyr::arrange(treatment_tissue) %>%
    tibble::column_to_rownames("treatment_tissue")
  
  # Ensure column order matches
  summary_matrix <- summary_matrix[, rownames(col_info), drop = FALSE]
  
  anno_colors <- create_annotation_colors(metadata)
  
  col_anno <- HeatmapAnnotation(
    Treatment = col_info$treatment,
    Tissue = col_info$tissue,
    col = list(
      Treatment = anno_colors$treatment,
      Tissue = anno_colors$tissue
    ),
    annotation_name_gp = gpar(fontsize = 10, fontface = "bold")
  )
  
  # Create color function
  if (metric == "mean") {
    max_val <- max(summary_matrix, na.rm = TRUE)
    col_fun <- colorRamp2(c(0, max_val/2, max_val), c("white", "pink", "red"))
  } else {
    col_fun <- colorRamp2(c(0, 50, 100), c("white", "orange", "red"))
  }
  
  # Create heatmap
  ht <- Heatmap(
    summary_matrix,
    name = legend_title,
    col = col_fun,
    
    # Appearance
    column_title = paste("S1PR", title_suffix, "by Treatment and Tissue"),
    column_title_gp = gpar(fontsize = 14, fontface = "bold"),
    row_title = "S1PR Genes", 
    row_title_gp = gpar(fontsize = 12, fontface = "bold"),
    row_names_gp = gpar(fontsize = 11, fontface = "italic"),
    column_names_gp = gpar(fontsize = 10),
    
    # Clustering
    cluster_rows = nrow(summary_matrix) > 1,
    cluster_columns = TRUE,
    
    # Annotations
    top_annotation = col_anno,
    
    # Cell text
    cell_fun = function(j, i, x, y, width, height, fill) {
      grid.text(sprintf("%.2f", summary_matrix[i, j]), 
                x, y, gp = gpar(fontsize = 9, col = "black"))
    },
    
    border = TRUE,
    heatmap_legend_param = list(
      title_gp = gpar(fontsize = 11, fontface = "bold"),
      labels_gp = gpar(fontsize = 9)
    )
  )
  
  return(list(
    heatmap = ht,
    summary_matrix = summary_matrix,
    summary_data = summary_data
  ))
}

# Function for detailed single-cell heatmap with subsampling by treatment and tissue
create_s1pr_detailed_treatment_tissue_heatmap <- function(seurat_obj, 
                                                          subsample_cells = 100,
                                                          scale_data = TRUE) {
  
  # Extract data
  s1pr_data <- extract_s1pr_data(seurat_obj)
  expr_data <- s1pr_data$expression
  metadata <- s1pr_data$metadata
  available_genes <- s1pr_data$available_genes
  
  # Subsample cells for visualization - stratified by treatment and tissue
  cat("Subsampling to", subsample_cells, "cells per treatment-tissue combination...\n")
  
  # Add cell names to metadata for easier manipulation
  metadata$cell_name <- rownames(metadata)
  
  # Stratified sampling by treatment and tissue - fixed the pull issue
  sampled_cells <- metadata %>%
    dplyr::mutate(treatment_tissue = paste(treatment, tissue, sep = "_")) %>%
    dplyr::group_by(treatment_tissue) %>%
    dplyr::sample_n(min(n(), subsample_cells)) %>%
    dplyr::ungroup() %>%
    dplyr::pull(cell_name)  # Now pulling a specific column
  
  expr_data <- expr_data[, sampled_cells]
  metadata <- metadata[sampled_cells, ]
  
  # Scale data if requested
  if (scale_data) {
    expr_data <- t(scale(t(expr_data)))
    # Handle any NaN values (genes with zero variance)
    expr_data[is.nan(expr_data)] <- 0
  }
  
  # Create annotation colors
  anno_colors <- create_annotation_colors(metadata)
  
  # Create column annotations
  col_anno <- HeatmapAnnotation(
    Treatment = metadata$treatment,
    Tissue = metadata$tissue,
    col = list(
      Treatment = anno_colors$treatment,
      Tissue = anno_colors$tissue
    ),
    annotation_name_gp = gpar(fontsize = 10, fontface = "bold")
  )
  
  # Order cells by treatment and tissue for better visualization
  metadata <- metadata %>%
    dplyr::mutate(treatment_tissue = paste(treatment, tissue, sep = "_"))
  cell_order <- order(metadata$treatment_tissue)
  expr_data_ordered <- expr_data[, cell_order]
  
  # Create color function for expression
  if (scale_data) {
    col_fun <- colorRamp2(c(-2, 0, 2), c("blue", "white", "red"))
    heatmap_title <- "S1PR Expression (Z-score) by Treatment and Tissue"
    legend_title <- "Z-score"
  } else {
    max_expr <- max(expr_data, na.rm = TRUE)
    col_fun <- colorRamp2(c(0, max_expr/2, max_expr), c("white", "pink", "red"))
    heatmap_title <- "S1PR Expression (Log-normalized) by Treatment and Tissue"
    legend_title <- "Expression"
  }
  
  # Create main heatmap
  ht <- Heatmap(
    expr_data_ordered,
    name = legend_title,
    col = col_fun,
    
    # Row parameters
    row_title = "S1PR Genes",
    row_title_gp = gpar(fontsize = 12, fontface = "bold"),
    row_names_gp = gpar(fontsize = 11, fontface = "italic"),
    cluster_rows = length(available_genes) > 1,
    
    # Column parameters  
    column_title = heatmap_title,
    column_title_gp = gpar(fontsize = 14, fontface = "bold"),
    cluster_columns = FALSE,
    show_column_names = FALSE,
    
    # Annotations
    top_annotation = col_anno,
    
    # Appearance
    border = TRUE,
    heatmap_legend_param = list(
      title_gp = gpar(fontsize = 11, fontface = "bold"),
      labels_gp = gpar(fontsize = 9)
    )
  )
  
  return(ht)
}

# Create directories if they don't exist
dir.create(here("R Projects", "2107claude", "output", "visualisation", "complex heatmaps"), 
           recursive = TRUE, showWarnings = FALSE)

# Create mean expression heatmap by treatment and tissue
cat("Creating S1PR mean expression heatmap by treatment and tissue...\n")
mean_heatmap_result <- create_s1pr_treatment_tissue_heatmap(seu_T, metric = "mean")
draw(mean_heatmap_result$heatmap)

# Save mean expression heatmap as SVG using a different approach
cat("Saving mean expression heatmap...\n")
heatmap_file <- here("R Projects", "2107claude", "output", "visualisation", "complex heatmaps", "s1pr_mean_expression_treatment_tissue_heatmap.svg")
svg(heatmap_file, width = 10, height = 6)
draw(mean_heatmap_result$heatmap)
dev.off()
cat("Mean expression heatmap saved to:", heatmap_file, "\n")

# Create percentage positive heatmap by treatment and tissue
cat("Creating S1PR percentage positive heatmap by treatment and tissue...\n")
pct_heatmap_result <- create_s1pr_treatment_tissue_heatmap(seu_T, metric = "pct_positive")
draw(pct_heatmap_result$heatmap)

# Save percentage positive heatmap as SVG
cat("Saving percentage positive heatmap...\n")
pct_file <- here("R Projects", "2107claude", "output", "visualisation", "complex heatmaps", "s1pr_percent_positive_treatment_tissue_heatmap.svg")
svg(pct_file, width = 10, height = 6)
draw(pct_heatmap_result$heatmap)
dev.off()
cat("Percentage positive heatmap saved to:", pct_file, "\n")

# Create detailed single-cell heatmap by treatment and tissue
cat("Creating detailed single-cell heatmap by treatment and tissue...\n")
detailed_heatmap <- create_s1pr_detailed_treatment_tissue_heatmap(seu_T, subsample_cells = 100, scale_data = TRUE)
draw(detailed_heatmap)

# Save detailed heatmap as SVG
cat("Saving detailed single-cell heatmap...\n")
detailed_file <- here("R Projects", "2107claude", "output", "visualisation", "complex heatmaps", "s1pr_detailed_treatment_tissue_heatmap.svg")
svg(detailed_file, width = 14, height = 6)
draw(detailed_heatmap)
dev.off()
cat("Detailed heatmap saved to:", detailed_file, "\n")

cat("All S1PR treatment-tissue heatmaps created and saved as SVG!\n")
```
